// @subframe/core/dist/cjs/components/radix/popover.js
"use client";
'use strict';

var index = require('../../packages/subframe-core/external/@radix-ui/react-popover/dist/index.js');

var Anchor = index.Anchor;
var Arrow = index.Arrow;
var Close = index.Close;
var Content = index.Content;
var Popover = index.Popover;
var PopoverAnchor = index.PopoverAnchor;
var PopoverArrow = index.PopoverArrow;
var PopoverClose = index.PopoverClose;
var PopoverContent = index.PopoverContent;
var PopoverPortal = index.PopoverPortal;
var PopoverTrigger = index.PopoverTrigger;
var Portal = index.Portal;
var Root = index.Root;
var Trigger = index.Trigger;

exports.Anchor = Anchor;
exports.Arrow = Arrow;
exports.Close = Close;
exports.Content = Content;
exports.Popover = Popover;
exports.PopoverAnchor = PopoverAnchor;
exports.PopoverArrow = PopoverArrow;
exports.PopoverClose = PopoverClose;
exports.PopoverContent = PopoverContent;
exports.PopoverPortal = PopoverPortal;
exports.PopoverTrigger = PopoverTrigger;
exports.Portal = Portal;
exports.Root = Root;
exports.Trigger = Trigger;
