// @subframe/core/dist/esm/components/radix/dropdown-menu.js
"use client";
import { Arrow as Arrow2, CheckboxItem as CheckboxItem2, Content as Content2, DropdownMenu as DropdownMenu$1, DropdownMenuArrow as DropdownMenuArrow$1, DropdownMenuCheckboxItem as DropdownMenuCheckboxItem$1, DropdownMenuContent as DropdownMenuContent$1, DropdownMenuGroup as DropdownMenuGroup$1, DropdownMenuItem as DropdownMenuItem$1, DropdownMenuItemIndicator as DropdownMenuItemIndicator$1, DropdownMenuLabel as DropdownMenuLabel$1, DropdownMenuPortal as DropdownMenuPortal$1, DropdownMenuRadioGroup as DropdownMenuRadioGroup$1, DropdownMenuRadioItem as DropdownMenuRadioItem$1, DropdownMenuSeparator as DropdownMenuSeparator$1, DropdownMenuSub as DropdownMenuSub$1, DropdownMenuSubContent as DropdownMenuSubContent$1, DropdownMenuSubTrigger as DropdownMenuSubTrigger$1, DropdownMenuTrigger as DropdownMenuTrigger$1, Group as Group2, Item as Item2, ItemIndicator as ItemIndicator2, Label as Label2, Portal as Portal2, RadioGroup as RadioGroup2, RadioItem as RadioItem2, Root as Root2, Separator as Separator2, Sub as Sub2, SubContent as SubContent2, SubTrigger as SubTrigger2, Trigger as Trigger$1 } from '../../packages/subframe-core/external/@radix-ui/react-dropdown-menu/dist/index.js';

var Arrow = Arrow2;
var CheckboxItem = CheckboxItem2;
var Content = Content2;
var DropdownMenu = DropdownMenu$1;
var DropdownMenuArrow = DropdownMenuArrow$1;
var DropdownMenuCheckboxItem = DropdownMenuCheckboxItem$1;
var DropdownMenuContent = DropdownMenuContent$1;
var DropdownMenuGroup = DropdownMenuGroup$1;
var DropdownMenuItem = DropdownMenuItem$1;
var DropdownMenuItemIndicator = DropdownMenuItemIndicator$1;
var DropdownMenuLabel = DropdownMenuLabel$1;
var DropdownMenuPortal = DropdownMenuPortal$1;
var DropdownMenuRadioGroup = DropdownMenuRadioGroup$1;
var DropdownMenuRadioItem = DropdownMenuRadioItem$1;
var DropdownMenuSeparator = DropdownMenuSeparator$1;
var DropdownMenuSub = DropdownMenuSub$1;
var DropdownMenuSubContent = DropdownMenuSubContent$1;
var DropdownMenuSubTrigger = DropdownMenuSubTrigger$1;
var DropdownMenuTrigger = DropdownMenuTrigger$1;
var Group = Group2;
var Item = Item2;
var ItemIndicator = ItemIndicator2;
var Label = Label2;
var Portal = Portal2;
var RadioGroup = RadioGroup2;
var RadioItem = RadioItem2;
var Root = Root2;
var Separator = Separator2;
var Sub = Sub2;
var SubContent = SubContent2;
var SubTrigger = SubTrigger2;
var Trigger = Trigger$1;

export { Arrow, CheckboxItem, Content, DropdownMenu, DropdownMenuArrow, DropdownMenuCheckboxItem, DropdownMenuContent, DropdownMenuGroup, DropdownMenuItem, DropdownMenuItemIndicator, DropdownMenuLabel, DropdownMenuPortal, DropdownMenuRadioGroup, DropdownMenuRadioItem, DropdownMenuSeparator, DropdownMenuSub, DropdownMenuSubContent, DropdownMenuSubTrigger, DropdownMenuTrigger, Group, Item, ItemIndicator, Label, Portal, RadioGroup, RadioItem, Root, Separator, Sub, SubContent, SubTrigger, Trigger };
