// @subframe/core/dist/esm/components/radix/popover.js
"use client";
import { Anchor as Anchor2, Arrow as Arrow2, Close as Close$1, Content as Content2, Popover as Popover$1, PopoverAnchor as PopoverAnchor$1, PopoverArrow as PopoverArrow$1, PopoverClose as PopoverClose$1, PopoverContent as PopoverContent$1, PopoverPortal as PopoverPortal$1, PopoverTrigger as PopoverTrigger$1, Portal as Portal$1, Root as Root2, Trigger as Trigger$1 } from '../../packages/subframe-core/external/@radix-ui/react-popover/dist/index.js';

var Anchor = Anchor2;
var Arrow = Arrow2;
var Close = Close$1;
var Content = Content2;
var Popover = Popover$1;
var PopoverAnchor = PopoverAnchor$1;
var PopoverArrow = PopoverArrow$1;
var PopoverClose = PopoverClose$1;
var PopoverContent = PopoverContent$1;
var PopoverPortal = PopoverPortal$1;
var PopoverTrigger = PopoverTrigger$1;
var Portal = Portal$1;
var Root = Root2;
var Trigger = Trigger$1;

export { Anchor, Arrow, Close, Content, Popover, PopoverAnchor, PopoverArrow, PopoverClose, PopoverContent, PopoverPortal, PopoverTrigger, Portal, Root, Trigger };
