// @subframe/core/dist/cjs/components/collapsible.js
"use client";
'use strict';

var _tslib = require('../_virtual/_tslib.js');
var index$2 = require('../external/@radix-ui/react-collapsible/dist/index.js');
var index = require('../packages/subframe-core/external/@radix-ui/react-slot/dist/index.js');
var index$1 = require('../external/classnames/index.js');
var React = require('react');
var collapsible_module = require('./collapsible.module.css.js');

var Chevron = React.forwardRef(function CollapsibleChevron(_a, ref) {
    var className = _a.className, otherProps = _tslib.__rest(_a, ["className"]);
    return React.createElement(index.Slot, _tslib.__assign({ ref: ref, className: index$1(className, collapsible_module.chevron) }, otherProps));
});
var Trigger = React.forwardRef(function Trigger(_a, ref) {
    var className = _a.className, children = _a.children, otherProps = _tslib.__rest(_a, ["className", "children"]);
    return (React.createElement(index$2.Trigger, _tslib.__assign({ className: index$1(className, collapsible_module.trigger), ref: ref }, otherProps), children));
});
var Content = React.forwardRef(function Content(_a, ref) {
    var className = _a.className, children = _a.children, otherProps = _tslib.__rest(_a, ["className", "children"]);
    return (React.createElement(index$2.Content, _tslib.__assign({ className: index$1(className, collapsible_module.content), ref: ref }, otherProps), children));
});
var Root = index$2.Root;
var Collapsible = {
    Root: Root,
    Trigger: Trigger,
    Content: Content,
    Chevron: Chevron,
};

exports.Chevron = Chevron;
exports.Collapsible = Collapsible;
exports.Content = Content;
exports.Root = Root;
exports.Trigger = Trigger;
