// @subframe/core/dist/cjs/components/select.js
"use client";
'use strict';

var _tslib = require('../_virtual/_tslib.js');
var index = require('../external/@radix-ui/react-select/dist/index.js');
var index$1 = require('../external/classnames/index.js');
var React = require('react');
var select_module = require('./select.module.css.js');

var Root = React.forwardRef(function SelectRoot(_a, _ref) {
    var children = _a.children, otherProps = _tslib.__rest(_a, ["children"]);
    return React.createElement(index.Root, _tslib.__assign({}, otherProps), children);
});
var Trigger = React.forwardRef(function Trigger(_a, ref) {
    var className = _a.className, children = _a.children, otherProps = _tslib.__rest(_a, ["className", "children"]);
    return (React.createElement(index.Trigger, _tslib.__assign({ className: index$1(className, select_module.trigger), ref: ref }, otherProps), children));
});
var Value = React.forwardRef(function Value(_a, ref) {
    _a.children; // NOTE: ignored
    var placeholder = _a.placeholder, className = _a.className, otherProps = _tslib.__rest(_a, ["children", "placeholder", "className"]);
    return (
    // NOTE: this needs a wrapper since Radix strips all styling from RadixSelect.Value
    React.createElement("span", _tslib.__assign({ className: index$1(className, select_module.value) }, otherProps, { ref: ref }),
        React.createElement(index.Value, { placeholder: placeholder })));
});
var Content = React.forwardRef(function Content(_a, ref) {
    var className = _a.className, children = _a.children, _b = _a.position, position = _b === void 0 ? "popper" : _b, _c = _a.sideOffset, sideOffset = _c === void 0 ? 4 : _c, otherProps = _tslib.__rest(_a, ["className", "children", "position", "sideOffset"]);
    return (React.createElement(index.Portal, null,
        React.createElement(index.Content, _tslib.__assign({ className: index$1(className, select_module.content), ref: ref, position: position, sideOffset: sideOffset }, otherProps),
            React.createElement(Select.Viewport, { asChild: true }, children))));
});
var Viewport = React.forwardRef(function Viewport(_a, ref) {
    var className = _a.className, children = _a.children, otherProps = _tslib.__rest(_a, ["className", "children"]);
    return (React.createElement(index.Viewport, _tslib.__assign({ className: index$1(className, select_module.viewport), ref: ref }, otherProps), children));
});
var Item = React.forwardRef(function Item(_a, ref) {
    var className = _a.className, children = _a.children, value = _a.value, otherProps = _tslib.__rest(_a, ["className", "children", "value"]);
    return (React.createElement(index.Item, _tslib.__assign({ className: index$1(className, select_module.item), ref: ref, value: value }, otherProps), children));
});
var ItemText = index.ItemText;
var Select = {
    Root: Root,
    Trigger: Trigger,
    Value: Value,
    Content: Content,
    Viewport: Viewport,
    Item: Item,
    ItemText: ItemText,
};

exports.Content = Content;
exports.Item = Item;
exports.ItemText = ItemText;
exports.Root = Root;
exports.Select = Select;
exports.Trigger = Trigger;
exports.Value = Value;
exports.Viewport = Viewport;
