// @subframe/core/dist/cjs/components/slider.js
"use client";
'use strict';

var _tslib = require('../_virtual/_tslib.js');
var index = require('../external/@radix-ui/react-slider/dist/index.js');
var index$1 = require('../external/classnames/index.js');
var React = require('react');
var slider_module = require('./slider.module.css.js');

var Root = React.forwardRef(function ProgressRoot(_a, ref) {
    var className = _a.className, children = _a.children, _b = _a.defaultValue, defaultValue = _b === void 0 ? [27] : _b, otherProps = _tslib.__rest(_a, ["className", "children", "defaultValue"]);
    return (React.createElement(index.Root, _tslib.__assign({ ref: ref, className: index$1(className, slider_module.sliderRoot), defaultValue: defaultValue }, otherProps), children));
});
var Track = React.forwardRef(function Indicator(_a, ref) {
    var className = _a.className, children = _a.children, otherProps = _tslib.__rest(_a, ["className", "children"]);
    return (React.createElement(index.Track, _tslib.__assign({ className: index$1(className, slider_module.sliderTrack), ref: ref }, otherProps), children));
});
var Range = React.forwardRef(function Indicator(_a, ref) {
    var className = _a.className, otherProps = _tslib.__rest(_a, ["className"]);
    return React.createElement(index.Range, _tslib.__assign({ className: index$1(className, slider_module.sliderRange), ref: ref }, otherProps));
});
var Thumb = React.forwardRef(function Indicator(_a, ref) {
    var className = _a.className, otherProps = _tslib.__rest(_a, ["className"]);
    return React.createElement(index.Thumb, _tslib.__assign({ className: index$1(className, slider_module.sliderThumb), ref: ref }, otherProps));
});
var Slider = { Root: Root, Track: Track, Range: Range, Thumb: Thumb };

exports.Range = Range;
exports.Root = Root;
exports.Slider = Slider;
exports.Thumb = Thumb;
exports.Track = Track;
