// @subframe/core/dist/esm/index.js
import { __assign } from './_virtual/_tslib.js';
import * as checkbox from './components/radix/checkbox.js';
import * as contextMenu from './components/radix/context-menu.js';
import * as dropdownMenu from './components/radix/dropdown-menu.js';
import * as hoverCard from './components/radix/hover-card.js';
import * as popover from './components/radix/popover.js';
import * as radioGroup from './components/radix/radio-group.js';
import * as tooltip from './components/radix/tooltip.js';
import { Toaster as Oe, toast as ce } from './external/sonner/dist/index.js';
import * as progress from './components/progress.js';
import * as slider from './components/slider.js';
import * as select from './components/select.js';
import { AreaChart } from './components/area-chart.js';
import { BarChart } from './components/bar-chart.js';
import { LineChart } from './components/line-chart.js';
import { PieChart } from './components/pie-chart.js';
import { CartesianGrid, XAxis, YAxis, ChartTooltip, ChartLegend, ChartLabel } from './components/charts.js';
import { Calendar } from './components/calendar.js';
import { Skeleton } from './components/skeleton.js';
import * as copyToClipboard from './components/copy-to-clipboard.js';
import * as _switch from './components/switch.js';
import { Loader } from './components/loader.js';
import { Icon } from './components/icon.js';
import { IconWrapper } from './components/icon-wrapper.js';
import * as dialog from './components/dialog.js';
import * as drawer from './components/drawer.js';
import * as fullscreenDialog from './components/fullscreen-dialog.js';
import * as toggleGroup from './components/toggle-group.js';
import * as collapsible from './components/collapsible.js';
import { Text } from './utilities/text.js';
import * as typescriptHelpers from './lib/typescript-helpers.js';
import { createTwClassNames, twClassNames } from './lib/tw-classnames.js';
import * as index from './assets/icons/final/index.js';
export { default as FeatherAArrowDown } from './assets/icons/final/feather-a-arrow-down.js';
export { default as FeatherAArrowUp } from './assets/icons/final/feather-a-arrow-up.js';
export { default as FeatherALargeSmall } from './assets/icons/final/feather-a-large-small.js';
export { default as FeatherAccessibility } from './assets/icons/final/feather-accessibility.js';
export { default as FeatherActivitySquare } from './assets/icons/final/feather-activity-square.js';
export { default as FeatherActivity } from './assets/icons/final/feather-activity.js';
export { default as FeatherAirVent } from './assets/icons/final/feather-air-vent.js';
export { default as FeatherAirplay } from './assets/icons/final/feather-airplay.js';
export { default as FeatherAlarmCheck } from './assets/icons/final/feather-alarm-check.js';
export { default as FeatherAlarmClockOff } from './assets/icons/final/feather-alarm-clock-off.js';
export { default as FeatherAlarmClock } from './assets/icons/final/feather-alarm-clock.js';
export { default as FeatherAlarmMinus } from './assets/icons/final/feather-alarm-minus.js';
export { default as FeatherAlarmPlus } from './assets/icons/final/feather-alarm-plus.js';
export { default as FeatherAlarmSmoke } from './assets/icons/final/feather-alarm-smoke.js';
export { default as FeatherAlbum } from './assets/icons/final/feather-album.js';
export { default as FeatherAlertCircle } from './assets/icons/final/feather-alert-circle.js';
export { default as FeatherAlertOctagon } from './assets/icons/final/feather-alert-octagon.js';
export { default as FeatherAlertTriangle } from './assets/icons/final/feather-alert-triangle.js';
export { default as FeatherAlignCenterHorizontal } from './assets/icons/final/feather-align-center-horizontal.js';
export { default as FeatherAlignCenterVertical } from './assets/icons/final/feather-align-center-vertical.js';
export { default as FeatherAlignCenter } from './assets/icons/final/feather-align-center.js';
export { default as FeatherAlignEndHorizontal } from './assets/icons/final/feather-align-end-horizontal.js';
export { default as FeatherAlignEndVertical } from './assets/icons/final/feather-align-end-vertical.js';
export { default as FeatherAlignHBottomLeft } from './assets/icons/final/feather-align-h-bottom-left.js';
export { default as FeatherAlignHBottomRight } from './assets/icons/final/feather-align-h-bottom-right.js';
export { default as FeatherAlignHBottom } from './assets/icons/final/feather-align-h-bottom.js';
export { default as FeatherAlignHCenter } from './assets/icons/final/feather-align-h-center.js';
export { default as FeatherAlignHLeft } from './assets/icons/final/feather-align-h-left.js';
export { default as FeatherAlignHRight } from './assets/icons/final/feather-align-h-right.js';
export { default as FeatherAlignHTopLeft } from './assets/icons/final/feather-align-h-top-left.js';
export { default as FeatherAlignHTopRight } from './assets/icons/final/feather-align-h-top-right.js';
export { default as FeatherAlignHTop } from './assets/icons/final/feather-align-h-top.js';
export { default as FeatherAlignHorizontalDistributeCenter } from './assets/icons/final/feather-align-horizontal-distribute-center.js';
export { default as FeatherAlignHorizontalDistributeEnd } from './assets/icons/final/feather-align-horizontal-distribute-end.js';
export { default as FeatherAlignHorizontalDistributeStart } from './assets/icons/final/feather-align-horizontal-distribute-start.js';
export { default as FeatherAlignHorizontalJustifyCenter } from './assets/icons/final/feather-align-horizontal-justify-center.js';
export { default as FeatherAlignHorizontalJustifyEnd } from './assets/icons/final/feather-align-horizontal-justify-end.js';
export { default as FeatherAlignHorizontalJustifyStart } from './assets/icons/final/feather-align-horizontal-justify-start.js';
export { default as FeatherAlignHorizontalSpaceAround } from './assets/icons/final/feather-align-horizontal-space-around.js';
export { default as FeatherAlignHorizontalSpaceBetween } from './assets/icons/final/feather-align-horizontal-space-between.js';
export { default as FeatherAlignJustify } from './assets/icons/final/feather-align-justify.js';
export { default as FeatherAlignLeft } from './assets/icons/final/feather-align-left.js';
export { default as FeatherAlignRight } from './assets/icons/final/feather-align-right.js';
export { default as FeatherAlignStartHorizontal } from './assets/icons/final/feather-align-start-horizontal.js';
export { default as FeatherAlignStartVertical } from './assets/icons/final/feather-align-start-vertical.js';
export { default as FeatherAlignVBottomLeft } from './assets/icons/final/feather-align-v-bottom-left.js';
export { default as FeatherAlignVBottomRight } from './assets/icons/final/feather-align-v-bottom-right.js';
export { default as FeatherAlignVBottom } from './assets/icons/final/feather-align-v-bottom.js';
export { default as FeatherAlignVCenter } from './assets/icons/final/feather-align-v-center.js';
export { default as FeatherAlignVLeft } from './assets/icons/final/feather-align-v-left.js';
export { default as FeatherAlignVRight } from './assets/icons/final/feather-align-v-right.js';
export { default as FeatherAlignVTopLeft } from './assets/icons/final/feather-align-v-top-left.js';
export { default as FeatherAlignVTopRight } from './assets/icons/final/feather-align-v-top-right.js';
export { default as FeatherAlignVTop } from './assets/icons/final/feather-align-v-top.js';
export { default as FeatherAlignVerticalDistributeCenter } from './assets/icons/final/feather-align-vertical-distribute-center.js';
export { default as FeatherAlignVerticalDistributeEnd } from './assets/icons/final/feather-align-vertical-distribute-end.js';
export { default as FeatherAlignVerticalDistributeStart } from './assets/icons/final/feather-align-vertical-distribute-start.js';
export { default as FeatherAlignVerticalJustifyCenter } from './assets/icons/final/feather-align-vertical-justify-center.js';
export { default as FeatherAlignVerticalJustifyEnd } from './assets/icons/final/feather-align-vertical-justify-end.js';
export { default as FeatherAlignVerticalJustifyStart } from './assets/icons/final/feather-align-vertical-justify-start.js';
export { default as FeatherAlignVerticalSpaceAround } from './assets/icons/final/feather-align-vertical-space-around.js';
export { default as FeatherAlignVerticalSpaceBetween } from './assets/icons/final/feather-align-vertical-space-between.js';
export { default as FeatherAmbulance } from './assets/icons/final/feather-ambulance.js';
export { default as FeatherAmpersand } from './assets/icons/final/feather-ampersand.js';
export { default as FeatherAmpersands } from './assets/icons/final/feather-ampersands.js';
export { default as FeatherAmphora } from './assets/icons/final/feather-amphora.js';
export { default as FeatherAnchor } from './assets/icons/final/feather-anchor.js';
export { default as FeatherAngry } from './assets/icons/final/feather-angry.js';
export { default as FeatherAnnoyed } from './assets/icons/final/feather-annoyed.js';
export { default as FeatherAntenna } from './assets/icons/final/feather-antenna.js';
export { default as FeatherAnvil } from './assets/icons/final/feather-anvil.js';
export { default as FeatherAperture } from './assets/icons/final/feather-aperture.js';
export { default as FeatherAppWindowMac } from './assets/icons/final/feather-app-window-mac.js';
export { default as FeatherAppWindow } from './assets/icons/final/feather-app-window.js';
export { default as FeatherApple } from './assets/icons/final/feather-apple.js';
export { default as FeatherArchiveRestore } from './assets/icons/final/feather-archive-restore.js';
export { default as FeatherArchiveX } from './assets/icons/final/feather-archive-x.js';
export { default as FeatherArchive } from './assets/icons/final/feather-archive.js';
export { default as FeatherAreaChart } from './assets/icons/final/feather-area-chart.js';
export { default as FeatherArmchair } from './assets/icons/final/feather-armchair.js';
export { default as FeatherArrowBigDownDash } from './assets/icons/final/feather-arrow-big-down-dash.js';
export { default as FeatherArrowBigDown } from './assets/icons/final/feather-arrow-big-down.js';
export { default as FeatherArrowBigLeftDash } from './assets/icons/final/feather-arrow-big-left-dash.js';
export { default as FeatherArrowBigLeft } from './assets/icons/final/feather-arrow-big-left.js';
export { default as FeatherArrowBigRightDash } from './assets/icons/final/feather-arrow-big-right-dash.js';
export { default as FeatherArrowBigRight } from './assets/icons/final/feather-arrow-big-right.js';
export { default as FeatherArrowBigUpDash } from './assets/icons/final/feather-arrow-big-up-dash.js';
export { default as FeatherArrowBigUp } from './assets/icons/final/feather-arrow-big-up.js';
export { default as FeatherArrowDown01 } from './assets/icons/final/feather-arrow-down-0-1.js';
export { default as FeatherArrowDown10 } from './assets/icons/final/feather-arrow-down-1-0.js';
export { default as FeatherArrowDownAZ } from './assets/icons/final/feather-arrow-down-a-z.js';
export { default as FeatherArrowDownCircle } from './assets/icons/final/feather-arrow-down-circle.js';
export { default as FeatherArrowDownFromLine } from './assets/icons/final/feather-arrow-down-from-line.js';
export { default as FeatherArrowDownLeftFromCircle } from './assets/icons/final/feather-arrow-down-left-from-circle.js';
export { default as FeatherArrowDownLeftSquare } from './assets/icons/final/feather-arrow-down-left-square.js';
export { default as FeatherArrowDownLeft } from './assets/icons/final/feather-arrow-down-left.js';
export { default as FeatherArrowDownNarrowWide } from './assets/icons/final/feather-arrow-down-narrow-wide.js';
export { default as FeatherArrowDownRightFromCircle } from './assets/icons/final/feather-arrow-down-right-from-circle.js';
export { default as FeatherArrowDownRightSquare } from './assets/icons/final/feather-arrow-down-right-square.js';
export { default as FeatherArrowDownRight } from './assets/icons/final/feather-arrow-down-right.js';
export { default as FeatherArrowDownSquare } from './assets/icons/final/feather-arrow-down-square.js';
export { default as FeatherArrowDownToDot } from './assets/icons/final/feather-arrow-down-to-dot.js';
export { default as FeatherArrowDownToLine } from './assets/icons/final/feather-arrow-down-to-line.js';
export { default as FeatherArrowDownUp } from './assets/icons/final/feather-arrow-down-up.js';
export { default as FeatherArrowDownWideNarrow } from './assets/icons/final/feather-arrow-down-wide-narrow.js';
export { default as FeatherArrowDownZA } from './assets/icons/final/feather-arrow-down-z-a.js';
export { default as FeatherArrowDown } from './assets/icons/final/feather-arrow-down.js';
export { default as FeatherArrowLeftCircle } from './assets/icons/final/feather-arrow-left-circle.js';
export { default as FeatherArrowLeftFromLine } from './assets/icons/final/feather-arrow-left-from-line.js';
export { default as FeatherArrowLeftRight } from './assets/icons/final/feather-arrow-left-right.js';
export { default as FeatherArrowLeftSquare } from './assets/icons/final/feather-arrow-left-square.js';
export { default as FeatherArrowLeftToLine } from './assets/icons/final/feather-arrow-left-to-line.js';
export { default as FeatherArrowLeft } from './assets/icons/final/feather-arrow-left.js';
export { default as FeatherArrowRightCircle } from './assets/icons/final/feather-arrow-right-circle.js';
export { default as FeatherArrowRightFromLine } from './assets/icons/final/feather-arrow-right-from-line.js';
export { default as FeatherArrowRightLeft } from './assets/icons/final/feather-arrow-right-left.js';
export { default as FeatherArrowRightSquare } from './assets/icons/final/feather-arrow-right-square.js';
export { default as FeatherArrowRightToLine } from './assets/icons/final/feather-arrow-right-to-line.js';
export { default as FeatherArrowRight } from './assets/icons/final/feather-arrow-right.js';
export { default as FeatherArrowUp01 } from './assets/icons/final/feather-arrow-up-0-1.js';
export { default as FeatherArrowUp10 } from './assets/icons/final/feather-arrow-up-1-0.js';
export { default as FeatherArrowUpAZ } from './assets/icons/final/feather-arrow-up-a-z.js';
export { default as FeatherArrowUpCircle } from './assets/icons/final/feather-arrow-up-circle.js';
export { default as FeatherArrowUpDown } from './assets/icons/final/feather-arrow-up-down.js';
export { default as FeatherArrowUpFromDot } from './assets/icons/final/feather-arrow-up-from-dot.js';
export { default as FeatherArrowUpFromLine } from './assets/icons/final/feather-arrow-up-from-line.js';
export { default as FeatherArrowUpLeftFromCircle } from './assets/icons/final/feather-arrow-up-left-from-circle.js';
export { default as FeatherArrowUpLeftSquare } from './assets/icons/final/feather-arrow-up-left-square.js';
export { default as FeatherArrowUpLeft } from './assets/icons/final/feather-arrow-up-left.js';
export { default as FeatherArrowUpNarrowWide } from './assets/icons/final/feather-arrow-up-narrow-wide.js';
export { default as FeatherArrowUpRightFromCircle } from './assets/icons/final/feather-arrow-up-right-from-circle.js';
export { default as FeatherArrowUpRightSquare } from './assets/icons/final/feather-arrow-up-right-square.js';
export { default as FeatherArrowUpRight } from './assets/icons/final/feather-arrow-up-right.js';
export { default as FeatherArrowUpSquare } from './assets/icons/final/feather-arrow-up-square.js';
export { default as FeatherArrowUpToLine } from './assets/icons/final/feather-arrow-up-to-line.js';
export { default as FeatherArrowUpWideNarrow } from './assets/icons/final/feather-arrow-up-wide-narrow.js';
export { default as FeatherArrowUpZA } from './assets/icons/final/feather-arrow-up-z-a.js';
export { default as FeatherArrowUp } from './assets/icons/final/feather-arrow-up.js';
export { default as FeatherArrowsUpFromLine } from './assets/icons/final/feather-arrows-up-from-line.js';
export { default as FeatherAsterisk } from './assets/icons/final/feather-asterisk.js';
export { default as FeatherAtSign } from './assets/icons/final/feather-at-sign.js';
export { default as FeatherAtom } from './assets/icons/final/feather-atom.js';
export { default as FeatherAudioLines } from './assets/icons/final/feather-audio-lines.js';
export { default as FeatherAudioWaveform } from './assets/icons/final/feather-audio-waveform.js';
export { default as FeatherAward } from './assets/icons/final/feather-award.js';
export { default as FeatherAxe } from './assets/icons/final/feather-axe.js';
export { default as FeatherAxis3D } from './assets/icons/final/feather-axis-3d.js';
export { default as FeatherBaby } from './assets/icons/final/feather-baby.js';
export { default as FeatherBackpack } from './assets/icons/final/feather-backpack.js';
export { default as FeatherBadgeAlert } from './assets/icons/final/feather-badge-alert.js';
export { default as FeatherBadgeCent } from './assets/icons/final/feather-badge-cent.js';
export { default as FeatherBadgeCheck } from './assets/icons/final/feather-badge-check.js';
export { default as FeatherBadgeDollarSign } from './assets/icons/final/feather-badge-dollar-sign.js';
export { default as FeatherBadgeEuro } from './assets/icons/final/feather-badge-euro.js';
export { default as FeatherBadgeHelp } from './assets/icons/final/feather-badge-help.js';
export { default as FeatherBadgeIndianRupee } from './assets/icons/final/feather-badge-indian-rupee.js';
export { default as FeatherBadgeInfo } from './assets/icons/final/feather-badge-info.js';
export { default as FeatherBadgeJapaneseYen } from './assets/icons/final/feather-badge-japanese-yen.js';
export { default as FeatherBadgeMinus } from './assets/icons/final/feather-badge-minus.js';
export { default as FeatherBadgePercent } from './assets/icons/final/feather-badge-percent.js';
export { default as FeatherBadgePlus } from './assets/icons/final/feather-badge-plus.js';
export { default as FeatherBadgePoundSterling } from './assets/icons/final/feather-badge-pound-sterling.js';
export { default as FeatherBadgeRussianRuble } from './assets/icons/final/feather-badge-russian-ruble.js';
export { default as FeatherBadgeSwissFranc } from './assets/icons/final/feather-badge-swiss-franc.js';
export { default as FeatherBadgeX } from './assets/icons/final/feather-badge-x.js';
export { default as FeatherBadge } from './assets/icons/final/feather-badge.js';
export { default as FeatherBaggageClaim } from './assets/icons/final/feather-baggage-claim.js';
export { default as FeatherBan } from './assets/icons/final/feather-ban.js';
export { default as FeatherBanana } from './assets/icons/final/feather-banana.js';
export { default as FeatherBandage } from './assets/icons/final/feather-bandage.js';
export { default as FeatherBanknoteArrowDown } from './assets/icons/final/feather-banknote-arrow-down.js';
export { default as FeatherBanknoteArrowUp } from './assets/icons/final/feather-banknote-arrow-up.js';
export { default as FeatherBanknoteX } from './assets/icons/final/feather-banknote-x.js';
export { default as FeatherBanknote } from './assets/icons/final/feather-banknote.js';
export { default as FeatherBarChart2 } from './assets/icons/final/feather-bar-chart-2.js';
export { default as FeatherBarChart3 } from './assets/icons/final/feather-bar-chart-3.js';
export { default as FeatherBarChart4 } from './assets/icons/final/feather-bar-chart-4.js';
export { default as FeatherBarChartBig } from './assets/icons/final/feather-bar-chart-big.js';
export { default as FeatherBarChartHorizontalBig } from './assets/icons/final/feather-bar-chart-horizontal-big.js';
export { default as FeatherBarChartHorizontal } from './assets/icons/final/feather-bar-chart-horizontal.js';
export { default as FeatherBarChart } from './assets/icons/final/feather-bar-chart.js';
export { default as FeatherBarcode } from './assets/icons/final/feather-barcode.js';
export { default as FeatherBarrel } from './assets/icons/final/feather-barrel.js';
export { default as FeatherBaseline } from './assets/icons/final/feather-baseline.js';
export { default as FeatherBath } from './assets/icons/final/feather-bath.js';
export { default as FeatherBatteryCharging } from './assets/icons/final/feather-battery-charging.js';
export { default as FeatherBatteryFull } from './assets/icons/final/feather-battery-full.js';
export { default as FeatherBatteryLow } from './assets/icons/final/feather-battery-low.js';
export { default as FeatherBatteryMedium } from './assets/icons/final/feather-battery-medium.js';
export { default as FeatherBatteryPlus } from './assets/icons/final/feather-battery-plus.js';
export { default as FeatherBatteryWarning } from './assets/icons/final/feather-battery-warning.js';
export { default as FeatherBattery } from './assets/icons/final/feather-battery.js';
export { default as FeatherBeaker } from './assets/icons/final/feather-beaker.js';
export { default as FeatherBeanOff } from './assets/icons/final/feather-bean-off.js';
export { default as FeatherBean } from './assets/icons/final/feather-bean.js';
export { default as FeatherBedDouble } from './assets/icons/final/feather-bed-double.js';
export { default as FeatherBedSingle } from './assets/icons/final/feather-bed-single.js';
export { default as FeatherBed } from './assets/icons/final/feather-bed.js';
export { default as FeatherBeef } from './assets/icons/final/feather-beef.js';
export { default as FeatherBeerOff } from './assets/icons/final/feather-beer-off.js';
export { default as FeatherBeer } from './assets/icons/final/feather-beer.js';
export { default as FeatherBellDot } from './assets/icons/final/feather-bell-dot.js';
export { default as FeatherBellElectric } from './assets/icons/final/feather-bell-electric.js';
export { default as FeatherBellMinus } from './assets/icons/final/feather-bell-minus.js';
export { default as FeatherBellOff } from './assets/icons/final/feather-bell-off.js';
export { default as FeatherBellPlus } from './assets/icons/final/feather-bell-plus.js';
export { default as FeatherBellRing } from './assets/icons/final/feather-bell-ring.js';
export { default as FeatherBell } from './assets/icons/final/feather-bell.js';
export { default as FeatherBetweenHorizontalEnd } from './assets/icons/final/feather-between-horizontal-end.js';
export { default as FeatherBetweenHorizontalStart } from './assets/icons/final/feather-between-horizontal-start.js';
export { default as FeatherBetweenVerticalEnd } from './assets/icons/final/feather-between-vertical-end.js';
export { default as FeatherBetweenVerticalStart } from './assets/icons/final/feather-between-vertical-start.js';
export { default as FeatherBicepsFlexed } from './assets/icons/final/feather-biceps-flexed.js';
export { default as FeatherBike } from './assets/icons/final/feather-bike.js';
export { default as FeatherBinary } from './assets/icons/final/feather-binary.js';
export { default as FeatherBinoculars } from './assets/icons/final/feather-binoculars.js';
export { default as FeatherBiohazard } from './assets/icons/final/feather-biohazard.js';
export { default as FeatherBird } from './assets/icons/final/feather-bird.js';
export { default as FeatherBitcoin } from './assets/icons/final/feather-bitcoin.js';
export { default as FeatherBlend } from './assets/icons/final/feather-blend.js';
export { default as FeatherBlinds } from './assets/icons/final/feather-blinds.js';
export { default as FeatherBlocks } from './assets/icons/final/feather-blocks.js';
export { default as FeatherBluetoothConnected } from './assets/icons/final/feather-bluetooth-connected.js';
export { default as FeatherBluetoothOff } from './assets/icons/final/feather-bluetooth-off.js';
export { default as FeatherBluetoothSearching } from './assets/icons/final/feather-bluetooth-searching.js';
export { default as FeatherBluetooth } from './assets/icons/final/feather-bluetooth.js';
export { default as FeatherBold } from './assets/icons/final/feather-bold.js';
export { default as FeatherBolt } from './assets/icons/final/feather-bolt.js';
export { default as FeatherBomb } from './assets/icons/final/feather-bomb.js';
export { default as FeatherBone } from './assets/icons/final/feather-bone.js';
export { default as FeatherBookA } from './assets/icons/final/feather-book-a.js';
export { default as FeatherBookAlert } from './assets/icons/final/feather-book-alert.js';
export { default as FeatherBookAudio } from './assets/icons/final/feather-book-audio.js';
export { default as FeatherBookCheck } from './assets/icons/final/feather-book-check.js';
export { default as FeatherBookCopy } from './assets/icons/final/feather-book-copy.js';
export { default as FeatherBookDashed } from './assets/icons/final/feather-book-dashed.js';
export { default as FeatherBookDown } from './assets/icons/final/feather-book-down.js';
export { default as FeatherBookHeadphones } from './assets/icons/final/feather-book-headphones.js';
export { default as FeatherBookHeart } from './assets/icons/final/feather-book-heart.js';
export { default as FeatherBookImage } from './assets/icons/final/feather-book-image.js';
export { default as FeatherBookKey } from './assets/icons/final/feather-book-key.js';
export { default as FeatherBookLock } from './assets/icons/final/feather-book-lock.js';
export { default as FeatherBookMarked } from './assets/icons/final/feather-book-marked.js';
export { default as FeatherBookMinus } from './assets/icons/final/feather-book-minus.js';
export { default as FeatherBookOpenCheck } from './assets/icons/final/feather-book-open-check.js';
export { default as FeatherBookOpenText } from './assets/icons/final/feather-book-open-text.js';
export { default as FeatherBookOpen } from './assets/icons/final/feather-book-open.js';
export { default as FeatherBookPlus } from './assets/icons/final/feather-book-plus.js';
export { default as FeatherBookText } from './assets/icons/final/feather-book-text.js';
export { default as FeatherBookType } from './assets/icons/final/feather-book-type.js';
export { default as FeatherBookUp2 } from './assets/icons/final/feather-book-up-2.js';
export { default as FeatherBookUp } from './assets/icons/final/feather-book-up.js';
export { default as FeatherBookUser } from './assets/icons/final/feather-book-user.js';
export { default as FeatherBookX } from './assets/icons/final/feather-book-x.js';
export { default as FeatherBook } from './assets/icons/final/feather-book.js';
export { default as FeatherBookmarkCheck } from './assets/icons/final/feather-bookmark-check.js';
export { default as FeatherBookmarkMinus } from './assets/icons/final/feather-bookmark-minus.js';
export { default as FeatherBookmarkPlus } from './assets/icons/final/feather-bookmark-plus.js';
export { default as FeatherBookmarkX } from './assets/icons/final/feather-bookmark-x.js';
export { default as FeatherBookmark } from './assets/icons/final/feather-bookmark.js';
export { default as FeatherBoomBox } from './assets/icons/final/feather-boom-box.js';
export { default as FeatherBotMessageSquare } from './assets/icons/final/feather-bot-message-square.js';
export { default as FeatherBotOff } from './assets/icons/final/feather-bot-off.js';
export { default as FeatherBot } from './assets/icons/final/feather-bot.js';
export { default as FeatherBottleWine } from './assets/icons/final/feather-bottle-wine.js';
export { default as FeatherBowArrow } from './assets/icons/final/feather-bow-arrow.js';
export { default as FeatherBoxSelect } from './assets/icons/final/feather-box-select.js';
export { default as FeatherBox } from './assets/icons/final/feather-box.js';
export { default as FeatherBoxes } from './assets/icons/final/feather-boxes.js';
export { default as FeatherBraces } from './assets/icons/final/feather-braces.js';
export { default as FeatherBrackets } from './assets/icons/final/feather-brackets.js';
export { default as FeatherBrainCircuit } from './assets/icons/final/feather-brain-circuit.js';
export { default as FeatherBrainCog } from './assets/icons/final/feather-brain-cog.js';
export { default as FeatherBrain } from './assets/icons/final/feather-brain.js';
export { default as FeatherBrickWallFire } from './assets/icons/final/feather-brick-wall-fire.js';
export { default as FeatherBrickWall } from './assets/icons/final/feather-brick-wall.js';
export { default as FeatherBriefcaseBusiness } from './assets/icons/final/feather-briefcase-business.js';
export { default as FeatherBriefcaseConveyorBelt } from './assets/icons/final/feather-briefcase-conveyor-belt.js';
export { default as FeatherBriefcaseMedical } from './assets/icons/final/feather-briefcase-medical.js';
export { default as FeatherBriefcase } from './assets/icons/final/feather-briefcase.js';
export { default as FeatherBringToFront } from './assets/icons/final/feather-bring-to-front.js';
export { default as FeatherBrushCleaning } from './assets/icons/final/feather-brush-cleaning.js';
export { default as FeatherBrush } from './assets/icons/final/feather-brush.js';
export { default as FeatherBubbles } from './assets/icons/final/feather-bubbles.js';
export { default as FeatherBugOff } from './assets/icons/final/feather-bug-off.js';
export { default as FeatherBugPlay } from './assets/icons/final/feather-bug-play.js';
export { default as FeatherBug } from './assets/icons/final/feather-bug.js';
export { default as FeatherBuilding2 } from './assets/icons/final/feather-building-2.js';
export { default as FeatherBuilding } from './assets/icons/final/feather-building.js';
export { default as FeatherBusFront } from './assets/icons/final/feather-bus-front.js';
export { default as FeatherBus } from './assets/icons/final/feather-bus.js';
export { default as FeatherCableCar } from './assets/icons/final/feather-cable-car.js';
export { default as FeatherCable } from './assets/icons/final/feather-cable.js';
export { default as FeatherCakeSlice } from './assets/icons/final/feather-cake-slice.js';
export { default as FeatherCake } from './assets/icons/final/feather-cake.js';
export { default as FeatherCalculator } from './assets/icons/final/feather-calculator.js';
export { default as FeatherCalendar1 } from './assets/icons/final/feather-calendar-1.js';
export { default as FeatherCalendarArrowDown } from './assets/icons/final/feather-calendar-arrow-down.js';
export { default as FeatherCalendarArrowUp } from './assets/icons/final/feather-calendar-arrow-up.js';
export { default as FeatherCalendarCheck2 } from './assets/icons/final/feather-calendar-check-2.js';
export { default as FeatherCalendarCheck } from './assets/icons/final/feather-calendar-check.js';
export { default as FeatherCalendarClock } from './assets/icons/final/feather-calendar-clock.js';
export { default as FeatherCalendarCog } from './assets/icons/final/feather-calendar-cog.js';
export { default as FeatherCalendarDays } from './assets/icons/final/feather-calendar-days.js';
export { default as FeatherCalendarFold } from './assets/icons/final/feather-calendar-fold.js';
export { default as FeatherCalendarHeart } from './assets/icons/final/feather-calendar-heart.js';
export { default as FeatherCalendarMinus2 } from './assets/icons/final/feather-calendar-minus-2.js';
export { default as FeatherCalendarMinus } from './assets/icons/final/feather-calendar-minus.js';
export { default as FeatherCalendarOff } from './assets/icons/final/feather-calendar-off.js';
export { default as FeatherCalendarPlus2 } from './assets/icons/final/feather-calendar-plus-2.js';
export { default as FeatherCalendarPlus } from './assets/icons/final/feather-calendar-plus.js';
export { default as FeatherCalendarRange } from './assets/icons/final/feather-calendar-range.js';
export { default as FeatherCalendarSearch } from './assets/icons/final/feather-calendar-search.js';
export { default as FeatherCalendarSync } from './assets/icons/final/feather-calendar-sync.js';
export { default as FeatherCalendarX2 } from './assets/icons/final/feather-calendar-x-2.js';
export { default as FeatherCalendarX } from './assets/icons/final/feather-calendar-x.js';
export { default as FeatherCalendar } from './assets/icons/final/feather-calendar.js';
export { default as FeatherCameraOff } from './assets/icons/final/feather-camera-off.js';
export { default as FeatherCamera } from './assets/icons/final/feather-camera.js';
export { default as FeatherCandlestickChart } from './assets/icons/final/feather-candlestick-chart.js';
export { default as FeatherCandyCane } from './assets/icons/final/feather-candy-cane.js';
export { default as FeatherCandyOff } from './assets/icons/final/feather-candy-off.js';
export { default as FeatherCandy } from './assets/icons/final/feather-candy.js';
export { default as FeatherCannabis } from './assets/icons/final/feather-cannabis.js';
export { default as FeatherCaptionsOff } from './assets/icons/final/feather-captions-off.js';
export { default as FeatherCaptions } from './assets/icons/final/feather-captions.js';
export { default as FeatherCarFront } from './assets/icons/final/feather-car-front.js';
export { default as FeatherCarTaxiFront } from './assets/icons/final/feather-car-taxi-front.js';
export { default as FeatherCar } from './assets/icons/final/feather-car.js';
export { default as FeatherCaravan } from './assets/icons/final/feather-caravan.js';
export { default as FeatherCardSim } from './assets/icons/final/feather-card-sim.js';
export { default as FeatherCarrot } from './assets/icons/final/feather-carrot.js';
export { default as FeatherCaseLower } from './assets/icons/final/feather-case-lower.js';
export { default as FeatherCaseSensitive } from './assets/icons/final/feather-case-sensitive.js';
export { default as FeatherCaseUpper } from './assets/icons/final/feather-case-upper.js';
export { default as FeatherCassetteTape } from './assets/icons/final/feather-cassette-tape.js';
export { default as FeatherCast } from './assets/icons/final/feather-cast.js';
export { default as FeatherCastle } from './assets/icons/final/feather-castle.js';
export { default as FeatherCat } from './assets/icons/final/feather-cat.js';
export { default as FeatherCctv } from './assets/icons/final/feather-cctv.js';
export { default as FeatherChartArea } from './assets/icons/final/feather-chart-area.js';
export { default as FeatherChartBarBig } from './assets/icons/final/feather-chart-bar-big.js';
export { default as FeatherChartBarDecreasing } from './assets/icons/final/feather-chart-bar-decreasing.js';
export { default as FeatherChartBarIncreasing } from './assets/icons/final/feather-chart-bar-increasing.js';
export { default as FeatherChartBarStacked } from './assets/icons/final/feather-chart-bar-stacked.js';
export { default as FeatherChartBar } from './assets/icons/final/feather-chart-bar.js';
export { default as FeatherChartCandlestick } from './assets/icons/final/feather-chart-candlestick.js';
export { default as FeatherChartColumnBig } from './assets/icons/final/feather-chart-column-big.js';
export { default as FeatherChartColumnDecreasing } from './assets/icons/final/feather-chart-column-decreasing.js';
export { default as FeatherChartColumnIncreasing } from './assets/icons/final/feather-chart-column-increasing.js';
export { default as FeatherChartColumnStacked } from './assets/icons/final/feather-chart-column-stacked.js';
export { default as FeatherChartColumn } from './assets/icons/final/feather-chart-column.js';
export { default as FeatherChartGantt } from './assets/icons/final/feather-chart-gantt.js';
export { default as FeatherChartLine } from './assets/icons/final/feather-chart-line.js';
export { default as FeatherChartNetwork } from './assets/icons/final/feather-chart-network.js';
export { default as FeatherChartNoAxesColumnDecreasing } from './assets/icons/final/feather-chart-no-axes-column-decreasing.js';
export { default as FeatherChartNoAxesCombined } from './assets/icons/final/feather-chart-no-axes-combined.js';
export { default as FeatherChartPie } from './assets/icons/final/feather-chart-pie.js';
export { default as FeatherChartScatter } from './assets/icons/final/feather-chart-scatter.js';
export { default as FeatherChartSpline } from './assets/icons/final/feather-chart-spline.js';
export { default as FeatherCheckCheck } from './assets/icons/final/feather-check-check.js';
export { default as FeatherCheckCircle2 } from './assets/icons/final/feather-check-circle-2.js';
export { default as FeatherCheckCircle } from './assets/icons/final/feather-check-circle.js';
export { default as FeatherCheckLine } from './assets/icons/final/feather-check-line.js';
export { default as FeatherCheckSquare } from './assets/icons/final/feather-check-square.js';
export { default as FeatherCheck } from './assets/icons/final/feather-check.js';
export { default as FeatherChefHat } from './assets/icons/final/feather-chef-hat.js';
export { default as FeatherCherry } from './assets/icons/final/feather-cherry.js';
export { default as FeatherChevronDownCircle } from './assets/icons/final/feather-chevron-down-circle.js';
export { default as FeatherChevronDownSquare } from './assets/icons/final/feather-chevron-down-square.js';
export { default as FeatherChevronDown } from './assets/icons/final/feather-chevron-down.js';
export { default as FeatherChevronFirst } from './assets/icons/final/feather-chevron-first.js';
export { default as FeatherChevronLast } from './assets/icons/final/feather-chevron-last.js';
export { default as FeatherChevronLeftCircle } from './assets/icons/final/feather-chevron-left-circle.js';
export { default as FeatherChevronLeftSquare } from './assets/icons/final/feather-chevron-left-square.js';
export { default as FeatherChevronLeft } from './assets/icons/final/feather-chevron-left.js';
export { default as FeatherChevronRightCircle } from './assets/icons/final/feather-chevron-right-circle.js';
export { default as FeatherChevronRightSquare } from './assets/icons/final/feather-chevron-right-square.js';
export { default as FeatherChevronRight } from './assets/icons/final/feather-chevron-right.js';
export { default as FeatherChevronUpCircle } from './assets/icons/final/feather-chevron-up-circle.js';
export { default as FeatherChevronUpSquare } from './assets/icons/final/feather-chevron-up-square.js';
export { default as FeatherChevronUp } from './assets/icons/final/feather-chevron-up.js';
export { default as FeatherChevronsDownUp } from './assets/icons/final/feather-chevrons-down-up.js';
export { default as FeatherChevronsDown } from './assets/icons/final/feather-chevrons-down.js';
export { default as FeatherChevronsLeftRightEllipsis } from './assets/icons/final/feather-chevrons-left-right-ellipsis.js';
export { default as FeatherChevronsLeftRight } from './assets/icons/final/feather-chevrons-left-right.js';
export { default as FeatherChevronsLeft } from './assets/icons/final/feather-chevrons-left.js';
export { default as FeatherChevronsRightLeft } from './assets/icons/final/feather-chevrons-right-left.js';
export { default as FeatherChevronsRight } from './assets/icons/final/feather-chevrons-right.js';
export { default as FeatherChevronsUpDown } from './assets/icons/final/feather-chevrons-up-down.js';
export { default as FeatherChevronsUp } from './assets/icons/final/feather-chevrons-up.js';
export { default as FeatherChrome } from './assets/icons/final/feather-chrome.js';
export { default as FeatherChurch } from './assets/icons/final/feather-church.js';
export { default as FeatherCigaretteOff } from './assets/icons/final/feather-cigarette-off.js';
export { default as FeatherCigarette } from './assets/icons/final/feather-cigarette.js';
export { default as FeatherCircleArrowLeft } from './assets/icons/final/feather-circle-arrow-left.js';
export { default as FeatherCircleArrowRight } from './assets/icons/final/feather-circle-arrow-right.js';
export { default as FeatherCircleCheckBig } from './assets/icons/final/feather-circle-check-big.js';
export { default as FeatherCircleCheck } from './assets/icons/final/feather-circle-check.js';
export { default as FeatherCircleDashed } from './assets/icons/final/feather-circle-dashed.js';
export { default as FeatherCircleDollarSign } from './assets/icons/final/feather-circle-dollar-sign.js';
export { default as FeatherCircleDotDashed } from './assets/icons/final/feather-circle-dot-dashed.js';
export { default as FeatherCircleDot } from './assets/icons/final/feather-circle-dot.js';
export { default as FeatherCircleEllipsis } from './assets/icons/final/feather-circle-ellipsis.js';
export { default as FeatherCircleEqual } from './assets/icons/final/feather-circle-equal.js';
export { default as FeatherCircleFadingArrowUp } from './assets/icons/final/feather-circle-fading-arrow-up.js';
export { default as FeatherCircleFadingPlus } from './assets/icons/final/feather-circle-fading-plus.js';
export { default as FeatherCircleOff } from './assets/icons/final/feather-circle-off.js';
export { default as FeatherCirclePoundSterling } from './assets/icons/final/feather-circle-pound-sterling.js';
export { default as FeatherCirclePower } from './assets/icons/final/feather-circle-power.js';
export { default as FeatherCircleSlash2 } from './assets/icons/final/feather-circle-slash-2.js';
export { default as FeatherCircleSlash } from './assets/icons/final/feather-circle-slash.js';
export { default as FeatherCircleSlashed } from './assets/icons/final/feather-circle-slashed.js';
export { default as FeatherCircleSmall } from './assets/icons/final/feather-circle-small.js';
export { default as FeatherCircleStop } from './assets/icons/final/feather-circle-stop.js';
export { default as FeatherCircle } from './assets/icons/final/feather-circle.js';
export { default as FeatherCircuitBoard } from './assets/icons/final/feather-circuit-board.js';
export { default as FeatherCitrus } from './assets/icons/final/feather-citrus.js';
export { default as FeatherClapperboard } from './assets/icons/final/feather-clapperboard.js';
export { default as FeatherClipboardCheck } from './assets/icons/final/feather-clipboard-check.js';
export { default as FeatherClipboardCopy } from './assets/icons/final/feather-clipboard-copy.js';
export { default as FeatherClipboardEdit } from './assets/icons/final/feather-clipboard-edit.js';
export { default as FeatherClipboardList } from './assets/icons/final/feather-clipboard-list.js';
export { default as FeatherClipboardMinus } from './assets/icons/final/feather-clipboard-minus.js';
export { default as FeatherClipboardPaste } from './assets/icons/final/feather-clipboard-paste.js';
export { default as FeatherClipboardPenLine } from './assets/icons/final/feather-clipboard-pen-line.js';
export { default as FeatherClipboardPen } from './assets/icons/final/feather-clipboard-pen.js';
export { default as FeatherClipboardPlus } from './assets/icons/final/feather-clipboard-plus.js';
export { default as FeatherClipboardSignature } from './assets/icons/final/feather-clipboard-signature.js';
export { default as FeatherClipboardType } from './assets/icons/final/feather-clipboard-type.js';
export { default as FeatherClipboardX } from './assets/icons/final/feather-clipboard-x.js';
export { default as FeatherClipboard } from './assets/icons/final/feather-clipboard.js';
export { default as FeatherClock1 } from './assets/icons/final/feather-clock-1.js';
export { default as FeatherClock10 } from './assets/icons/final/feather-clock-10.js';
export { default as FeatherClock11 } from './assets/icons/final/feather-clock-11.js';
export { default as FeatherClock12 } from './assets/icons/final/feather-clock-12.js';
export { default as FeatherClock2 } from './assets/icons/final/feather-clock-2.js';
export { default as FeatherClock3 } from './assets/icons/final/feather-clock-3.js';
export { default as FeatherClock4 } from './assets/icons/final/feather-clock-4.js';
export { default as FeatherClock5 } from './assets/icons/final/feather-clock-5.js';
export { default as FeatherClock6 } from './assets/icons/final/feather-clock-6.js';
export { default as FeatherClock7 } from './assets/icons/final/feather-clock-7.js';
export { default as FeatherClock8 } from './assets/icons/final/feather-clock-8.js';
export { default as FeatherClock9 } from './assets/icons/final/feather-clock-9.js';
export { default as FeatherClockAlert } from './assets/icons/final/feather-clock-alert.js';
export { default as FeatherClockArrowDown } from './assets/icons/final/feather-clock-arrow-down.js';
export { default as FeatherClockArrowUp } from './assets/icons/final/feather-clock-arrow-up.js';
export { default as FeatherClockFading } from './assets/icons/final/feather-clock-fading.js';
export { default as FeatherClockPlus } from './assets/icons/final/feather-clock-plus.js';
export { default as FeatherClock } from './assets/icons/final/feather-clock.js';
export { default as FeatherCloudAlert } from './assets/icons/final/feather-cloud-alert.js';
export { default as FeatherCloudCheck } from './assets/icons/final/feather-cloud-check.js';
export { default as FeatherCloudCog } from './assets/icons/final/feather-cloud-cog.js';
export { default as FeatherCloudDownload } from './assets/icons/final/feather-cloud-download.js';
export { default as FeatherCloudDrizzle } from './assets/icons/final/feather-cloud-drizzle.js';
export { default as FeatherCloudFog } from './assets/icons/final/feather-cloud-fog.js';
export { default as FeatherCloudHail } from './assets/icons/final/feather-cloud-hail.js';
export { default as FeatherCloudLightning } from './assets/icons/final/feather-cloud-lightning.js';
export { default as FeatherCloudMoonRain } from './assets/icons/final/feather-cloud-moon-rain.js';
export { default as FeatherCloudMoon } from './assets/icons/final/feather-cloud-moon.js';
export { default as FeatherCloudOff } from './assets/icons/final/feather-cloud-off.js';
export { default as FeatherCloudRainWind } from './assets/icons/final/feather-cloud-rain-wind.js';
export { default as FeatherCloudRain } from './assets/icons/final/feather-cloud-rain.js';
export { default as FeatherCloudSnow } from './assets/icons/final/feather-cloud-snow.js';
export { default as FeatherCloudSunRain } from './assets/icons/final/feather-cloud-sun-rain.js';
export { default as FeatherCloudSun } from './assets/icons/final/feather-cloud-sun.js';
export { default as FeatherCloudUpload } from './assets/icons/final/feather-cloud-upload.js';
export { default as FeatherCloud } from './assets/icons/final/feather-cloud.js';
export { default as FeatherCloudy } from './assets/icons/final/feather-cloudy.js';
export { default as FeatherClover } from './assets/icons/final/feather-clover.js';
export { default as FeatherClub } from './assets/icons/final/feather-club.js';
export { default as FeatherCode2 } from './assets/icons/final/feather-code-2.js';
export { default as FeatherCode } from './assets/icons/final/feather-code.js';
export { default as FeatherCodepen } from './assets/icons/final/feather-codepen.js';
export { default as FeatherCodesandbox } from './assets/icons/final/feather-codesandbox.js';
export { default as FeatherCoffee } from './assets/icons/final/feather-coffee.js';
export { default as FeatherCog } from './assets/icons/final/feather-cog.js';
export { default as FeatherCoins } from './assets/icons/final/feather-coins.js';
export { default as FeatherColumns2 } from './assets/icons/final/feather-columns-2.js';
export { default as FeatherColumns3Cog } from './assets/icons/final/feather-columns-3-cog.js';
export { default as FeatherColumns3 } from './assets/icons/final/feather-columns-3.js';
export { default as FeatherColumns4 } from './assets/icons/final/feather-columns-4.js';
export { default as FeatherColumns } from './assets/icons/final/feather-columns.js';
export { default as FeatherCombine } from './assets/icons/final/feather-combine.js';
export { default as FeatherCommand } from './assets/icons/final/feather-command.js';
export { default as FeatherCompass } from './assets/icons/final/feather-compass.js';
export { default as FeatherComponent } from './assets/icons/final/feather-component.js';
export { default as FeatherComputer } from './assets/icons/final/feather-computer.js';
export { default as FeatherConciergeBell } from './assets/icons/final/feather-concierge-bell.js';
export { default as FeatherCone } from './assets/icons/final/feather-cone.js';
export { default as FeatherConstruction } from './assets/icons/final/feather-construction.js';
export { default as FeatherContact2 } from './assets/icons/final/feather-contact-2.js';
export { default as FeatherContactRound } from './assets/icons/final/feather-contact-round.js';
export { default as FeatherContact } from './assets/icons/final/feather-contact.js';
export { default as FeatherContainer } from './assets/icons/final/feather-container.js';
export { default as FeatherContrast } from './assets/icons/final/feather-contrast.js';
export { default as FeatherCookie } from './assets/icons/final/feather-cookie.js';
export { default as FeatherCookingPot } from './assets/icons/final/feather-cooking-pot.js';
export { default as FeatherCopyCheck } from './assets/icons/final/feather-copy-check.js';
export { default as FeatherCopyMinus } from './assets/icons/final/feather-copy-minus.js';
export { default as FeatherCopyPlus } from './assets/icons/final/feather-copy-plus.js';
export { default as FeatherCopySlash } from './assets/icons/final/feather-copy-slash.js';
export { default as FeatherCopyX } from './assets/icons/final/feather-copy-x.js';
export { default as FeatherCopy } from './assets/icons/final/feather-copy.js';
export { default as FeatherCopyleft } from './assets/icons/final/feather-copyleft.js';
export { default as FeatherCopyright } from './assets/icons/final/feather-copyright.js';
export { default as FeatherCornerDownLeft } from './assets/icons/final/feather-corner-down-left.js';
export { default as FeatherCornerDownRight } from './assets/icons/final/feather-corner-down-right.js';
export { default as FeatherCornerFull } from './assets/icons/final/feather-corner-full.js';
export { default as FeatherCornerLarge } from './assets/icons/final/feather-corner-large.js';
export { default as FeatherCornerLeftDown } from './assets/icons/final/feather-corner-left-down.js';
export { default as FeatherCornerLeftUp } from './assets/icons/final/feather-corner-left-up.js';
export { default as FeatherCornerMedium } from './assets/icons/final/feather-corner-medium.js';
export { default as FeatherCornerRightDown } from './assets/icons/final/feather-corner-right-down.js';
export { default as FeatherCornerRightUp } from './assets/icons/final/feather-corner-right-up.js';
export { default as FeatherCornerSmall } from './assets/icons/final/feather-corner-small.js';
export { default as FeatherCornerUpLeft } from './assets/icons/final/feather-corner-up-left.js';
export { default as FeatherCornerUpRight } from './assets/icons/final/feather-corner-up-right.js';
export { default as FeatherCpu } from './assets/icons/final/feather-cpu.js';
export { default as FeatherCreativeCommons } from './assets/icons/final/feather-creative-commons.js';
export { default as FeatherCreditCard } from './assets/icons/final/feather-credit-card.js';
export { default as FeatherCroissant } from './assets/icons/final/feather-croissant.js';
export { default as FeatherCrop } from './assets/icons/final/feather-crop.js';
export { default as FeatherCross } from './assets/icons/final/feather-cross.js';
export { default as FeatherCrosshair } from './assets/icons/final/feather-crosshair.js';
export { default as FeatherCrown } from './assets/icons/final/feather-crown.js';
export { default as FeatherCuboid } from './assets/icons/final/feather-cuboid.js';
export { default as FeatherCupSoda } from './assets/icons/final/feather-cup-soda.js';
export { default as FeatherCurlyBraces } from './assets/icons/final/feather-curly-braces.js';
export { default as FeatherCurrency } from './assets/icons/final/feather-currency.js';
export { default as FeatherCylinder } from './assets/icons/final/feather-cylinder.js';
export { default as FeatherDam } from './assets/icons/final/feather-dam.js';
export { default as FeatherDatabaseBackup } from './assets/icons/final/feather-database-backup.js';
export { default as FeatherDatabaseZap } from './assets/icons/final/feather-database-zap.js';
export { default as FeatherDatabase } from './assets/icons/final/feather-database.js';
export { default as FeatherDecimalsArrowLeft } from './assets/icons/final/feather-decimals-arrow-left.js';
export { default as FeatherDecimalsArrowRight } from './assets/icons/final/feather-decimals-arrow-right.js';
export { default as FeatherDelete } from './assets/icons/final/feather-delete.js';
export { default as FeatherDessert } from './assets/icons/final/feather-dessert.js';
export { default as FeatherDiameter } from './assets/icons/final/feather-diameter.js';
export { default as FeatherDiamondMinus } from './assets/icons/final/feather-diamond-minus.js';
export { default as FeatherDiamondPlus } from './assets/icons/final/feather-diamond-plus.js';
export { default as FeatherDiamond } from './assets/icons/final/feather-diamond.js';
export { default as FeatherDice1 } from './assets/icons/final/feather-dice-1.js';
export { default as FeatherDice2 } from './assets/icons/final/feather-dice-2.js';
export { default as FeatherDice3 } from './assets/icons/final/feather-dice-3.js';
export { default as FeatherDice4 } from './assets/icons/final/feather-dice-4.js';
export { default as FeatherDice5 } from './assets/icons/final/feather-dice-5.js';
export { default as FeatherDice6 } from './assets/icons/final/feather-dice-6.js';
export { default as FeatherDices } from './assets/icons/final/feather-dices.js';
export { default as FeatherDiff } from './assets/icons/final/feather-diff.js';
export { default as FeatherDisc2 } from './assets/icons/final/feather-disc-2.js';
export { default as FeatherDisc3 } from './assets/icons/final/feather-disc-3.js';
export { default as FeatherDiscAlbum } from './assets/icons/final/feather-disc-album.js';
export { default as FeatherDisc } from './assets/icons/final/feather-disc.js';
export { default as FeatherDivideCircle } from './assets/icons/final/feather-divide-circle.js';
export { default as FeatherDivideSquare } from './assets/icons/final/feather-divide-square.js';
export { default as FeatherDivide } from './assets/icons/final/feather-divide.js';
export { default as FeatherDnaOff } from './assets/icons/final/feather-dna-off.js';
export { default as FeatherDna } from './assets/icons/final/feather-dna.js';
export { default as FeatherDock } from './assets/icons/final/feather-dock.js';
export { default as FeatherDog } from './assets/icons/final/feather-dog.js';
export { default as FeatherDollarSign } from './assets/icons/final/feather-dollar-sign.js';
export { default as FeatherDonut } from './assets/icons/final/feather-donut.js';
export { default as FeatherDoorClosedLocked } from './assets/icons/final/feather-door-closed-locked.js';
export { default as FeatherDoorClosed } from './assets/icons/final/feather-door-closed.js';
export { default as FeatherDoorOpen } from './assets/icons/final/feather-door-open.js';
export { default as FeatherDot } from './assets/icons/final/feather-dot.js';
export { default as FeatherDownloadCloud } from './assets/icons/final/feather-download-cloud.js';
export { default as FeatherDownload } from './assets/icons/final/feather-download.js';
export { default as FeatherDraftingCompass } from './assets/icons/final/feather-drafting-compass.js';
export { default as FeatherDrama } from './assets/icons/final/feather-drama.js';
export { default as FeatherDribbble } from './assets/icons/final/feather-dribbble.js';
export { default as FeatherDrill } from './assets/icons/final/feather-drill.js';
export { default as FeatherDrone } from './assets/icons/final/feather-drone.js';
export { default as FeatherDropletOff } from './assets/icons/final/feather-droplet-off.js';
export { default as FeatherDroplet } from './assets/icons/final/feather-droplet.js';
export { default as FeatherDroplets } from './assets/icons/final/feather-droplets.js';
export { default as FeatherDrum } from './assets/icons/final/feather-drum.js';
export { default as FeatherDrumstick } from './assets/icons/final/feather-drumstick.js';
export { default as FeatherDumbbell } from './assets/icons/final/feather-dumbbell.js';
export { default as FeatherEarOff } from './assets/icons/final/feather-ear-off.js';
export { default as FeatherEar } from './assets/icons/final/feather-ear.js';
export { default as FeatherEarthLock } from './assets/icons/final/feather-earth-lock.js';
export { default as FeatherEarth } from './assets/icons/final/feather-earth.js';
export { default as FeatherEclipse } from './assets/icons/final/feather-eclipse.js';
export { default as FeatherEdit2 } from './assets/icons/final/feather-edit-2.js';
export { default as FeatherEdit3 } from './assets/icons/final/feather-edit-3.js';
export { default as FeatherEdit } from './assets/icons/final/feather-edit.js';
export { default as FeatherEggFried } from './assets/icons/final/feather-egg-fried.js';
export { default as FeatherEggOff } from './assets/icons/final/feather-egg-off.js';
export { default as FeatherEgg } from './assets/icons/final/feather-egg.js';
export { default as FeatherEqualApproximately } from './assets/icons/final/feather-equal-approximately.js';
export { default as FeatherEqualNot } from './assets/icons/final/feather-equal-not.js';
export { default as FeatherEqual } from './assets/icons/final/feather-equal.js';
export { default as FeatherEraser } from './assets/icons/final/feather-eraser.js';
export { default as FeatherEthernetPort } from './assets/icons/final/feather-ethernet-port.js';
export { default as FeatherEuro } from './assets/icons/final/feather-euro.js';
export { default as FeatherExpand } from './assets/icons/final/feather-expand.js';
export { default as FeatherExternalLink } from './assets/icons/final/feather-external-link.js';
export { default as FeatherEyeClosed } from './assets/icons/final/feather-eye-closed.js';
export { default as FeatherEyeOff } from './assets/icons/final/feather-eye-off.js';
export { default as FeatherEye } from './assets/icons/final/feather-eye.js';
export { default as FeatherFacebook } from './assets/icons/final/feather-facebook.js';
export { default as FeatherFactory } from './assets/icons/final/feather-factory.js';
export { default as FeatherFan } from './assets/icons/final/feather-fan.js';
export { default as FeatherFastForward } from './assets/icons/final/feather-fast-forward.js';
export { default as FeatherFeather } from './assets/icons/final/feather-feather.js';
export { default as FeatherFence } from './assets/icons/final/feather-fence.js';
export { default as FeatherFerrisWheel } from './assets/icons/final/feather-ferris-wheel.js';
export { default as FeatherFigma } from './assets/icons/final/feather-figma.js';
export { default as FeatherFileArchive } from './assets/icons/final/feather-file-archive.js';
export { default as FeatherFileAudio2 } from './assets/icons/final/feather-file-audio-2.js';
export { default as FeatherFileAudio } from './assets/icons/final/feather-file-audio.js';
export { default as FeatherFileAxis3D } from './assets/icons/final/feather-file-axis-3d.js';
export { default as FeatherFileBadge2 } from './assets/icons/final/feather-file-badge-2.js';
export { default as FeatherFileBadge } from './assets/icons/final/feather-file-badge.js';
export { default as FeatherFileBarChart2 } from './assets/icons/final/feather-file-bar-chart-2.js';
export { default as FeatherFileBarChart } from './assets/icons/final/feather-file-bar-chart.js';
export { default as FeatherFileBox } from './assets/icons/final/feather-file-box.js';
export { default as FeatherFileChartColumnIncreasing } from './assets/icons/final/feather-file-chart-column-increasing.js';
export { default as FeatherFileChartColumn } from './assets/icons/final/feather-file-chart-column.js';
export { default as FeatherFileChartLine } from './assets/icons/final/feather-file-chart-line.js';
export { default as FeatherFileChartPie } from './assets/icons/final/feather-file-chart-pie.js';
export { default as FeatherFileCheck2 } from './assets/icons/final/feather-file-check-2.js';
export { default as FeatherFileCheck } from './assets/icons/final/feather-file-check.js';
export { default as FeatherFileClock } from './assets/icons/final/feather-file-clock.js';
export { default as FeatherFileCode2 } from './assets/icons/final/feather-file-code-2.js';
export { default as FeatherFileCode } from './assets/icons/final/feather-file-code.js';
export { default as FeatherFileCog2 } from './assets/icons/final/feather-file-cog-2.js';
export { default as FeatherFileCog } from './assets/icons/final/feather-file-cog.js';
export { default as FeatherFileDiff } from './assets/icons/final/feather-file-diff.js';
export { default as FeatherFileDigit } from './assets/icons/final/feather-file-digit.js';
export { default as FeatherFileDown } from './assets/icons/final/feather-file-down.js';
export { default as FeatherFileEdit } from './assets/icons/final/feather-file-edit.js';
export { default as FeatherFileHeart } from './assets/icons/final/feather-file-heart.js';
export { default as FeatherFileImage } from './assets/icons/final/feather-file-image.js';
export { default as FeatherFileInput } from './assets/icons/final/feather-file-input.js';
export { default as FeatherFileJson2 } from './assets/icons/final/feather-file-json-2.js';
export { default as FeatherFileJson } from './assets/icons/final/feather-file-json.js';
export { default as FeatherFileKey2 } from './assets/icons/final/feather-file-key-2.js';
export { default as FeatherFileKey } from './assets/icons/final/feather-file-key.js';
export { default as FeatherFileLineChart } from './assets/icons/final/feather-file-line-chart.js';
export { default as FeatherFileLock2 } from './assets/icons/final/feather-file-lock-2.js';
export { default as FeatherFileLock } from './assets/icons/final/feather-file-lock.js';
export { default as FeatherFileMinus2 } from './assets/icons/final/feather-file-minus-2.js';
export { default as FeatherFileMinus } from './assets/icons/final/feather-file-minus.js';
export { default as FeatherFileMusic } from './assets/icons/final/feather-file-music.js';
export { default as FeatherFileOutput } from './assets/icons/final/feather-file-output.js';
export { default as FeatherFilePenLine } from './assets/icons/final/feather-file-pen-line.js';
export { default as FeatherFilePen } from './assets/icons/final/feather-file-pen.js';
export { default as FeatherFilePieChart } from './assets/icons/final/feather-file-pie-chart.js';
export { default as FeatherFilePlus2 } from './assets/icons/final/feather-file-plus-2.js';
export { default as FeatherFilePlus } from './assets/icons/final/feather-file-plus.js';
export { default as FeatherFileQuestionMark } from './assets/icons/final/feather-file-question-mark.js';
export { default as FeatherFileQuestion } from './assets/icons/final/feather-file-question.js';
export { default as FeatherFileScan } from './assets/icons/final/feather-file-scan.js';
export { default as FeatherFileSearch2 } from './assets/icons/final/feather-file-search-2.js';
export { default as FeatherFileSearch } from './assets/icons/final/feather-file-search.js';
export { default as FeatherFileSignature } from './assets/icons/final/feather-file-signature.js';
export { default as FeatherFileSliders } from './assets/icons/final/feather-file-sliders.js';
export { default as FeatherFileSpreadsheet } from './assets/icons/final/feather-file-spreadsheet.js';
export { default as FeatherFileStack } from './assets/icons/final/feather-file-stack.js';
export { default as FeatherFileSymlink } from './assets/icons/final/feather-file-symlink.js';
export { default as FeatherFileTerminal } from './assets/icons/final/feather-file-terminal.js';
export { default as FeatherFileText } from './assets/icons/final/feather-file-text.js';
export { default as FeatherFileType2 } from './assets/icons/final/feather-file-type-2.js';
export { default as FeatherFileType } from './assets/icons/final/feather-file-type.js';
export { default as FeatherFileUp } from './assets/icons/final/feather-file-up.js';
export { default as FeatherFileUser } from './assets/icons/final/feather-file-user.js';
export { default as FeatherFileVideo2 } from './assets/icons/final/feather-file-video-2.js';
export { default as FeatherFileVideo } from './assets/icons/final/feather-file-video.js';
export { default as FeatherFileVolume2 } from './assets/icons/final/feather-file-volume-2.js';
export { default as FeatherFileVolume } from './assets/icons/final/feather-file-volume.js';
export { default as FeatherFileWarning } from './assets/icons/final/feather-file-warning.js';
export { default as FeatherFileX2 } from './assets/icons/final/feather-file-x-2.js';
export { default as FeatherFileX } from './assets/icons/final/feather-file-x.js';
export { default as FeatherFile } from './assets/icons/final/feather-file.js';
export { default as FeatherFiles } from './assets/icons/final/feather-files.js';
export { default as FeatherFilm } from './assets/icons/final/feather-film.js';
export { default as FeatherFilterX } from './assets/icons/final/feather-filter-x.js';
export { default as FeatherFilter } from './assets/icons/final/feather-filter.js';
export { default as FeatherFingerprint } from './assets/icons/final/feather-fingerprint.js';
export { default as FeatherFireExtinguisher } from './assets/icons/final/feather-fire-extinguisher.js';
export { default as FeatherFishOff } from './assets/icons/final/feather-fish-off.js';
export { default as FeatherFishSymbol } from './assets/icons/final/feather-fish-symbol.js';
export { default as FeatherFish } from './assets/icons/final/feather-fish.js';
export { default as FeatherFlagOff } from './assets/icons/final/feather-flag-off.js';
export { default as FeatherFlagTriangleLeft } from './assets/icons/final/feather-flag-triangle-left.js';
export { default as FeatherFlagTriangleRight } from './assets/icons/final/feather-flag-triangle-right.js';
export { default as FeatherFlag } from './assets/icons/final/feather-flag.js';
export { default as FeatherFlameKindling } from './assets/icons/final/feather-flame-kindling.js';
export { default as FeatherFlame } from './assets/icons/final/feather-flame.js';
export { default as FeatherFlashlightOff } from './assets/icons/final/feather-flashlight-off.js';
export { default as FeatherFlashlight } from './assets/icons/final/feather-flashlight.js';
export { default as FeatherFlaskConicalOff } from './assets/icons/final/feather-flask-conical-off.js';
export { default as FeatherFlaskConical } from './assets/icons/final/feather-flask-conical.js';
export { default as FeatherFlaskRound } from './assets/icons/final/feather-flask-round.js';
export { default as FeatherFlipHorizontal2 } from './assets/icons/final/feather-flip-horizontal-2.js';
export { default as FeatherFlipHorizontal } from './assets/icons/final/feather-flip-horizontal.js';
export { default as FeatherFlipVertical2 } from './assets/icons/final/feather-flip-vertical-2.js';
export { default as FeatherFlipVertical } from './assets/icons/final/feather-flip-vertical.js';
export { default as FeatherFlower2 } from './assets/icons/final/feather-flower-2.js';
export { default as FeatherFlower } from './assets/icons/final/feather-flower.js';
export { default as FeatherFocus } from './assets/icons/final/feather-focus.js';
export { default as FeatherFoldHorizontal } from './assets/icons/final/feather-fold-horizontal.js';
export { default as FeatherFoldVertical } from './assets/icons/final/feather-fold-vertical.js';
export { default as FeatherFolderArchive } from './assets/icons/final/feather-folder-archive.js';
export { default as FeatherFolderCheck } from './assets/icons/final/feather-folder-check.js';
export { default as FeatherFolderClock } from './assets/icons/final/feather-folder-clock.js';
export { default as FeatherFolderClosed } from './assets/icons/final/feather-folder-closed.js';
export { default as FeatherFolderCode } from './assets/icons/final/feather-folder-code.js';
export { default as FeatherFolderCog2 } from './assets/icons/final/feather-folder-cog-2.js';
export { default as FeatherFolderCog } from './assets/icons/final/feather-folder-cog.js';
export { default as FeatherFolderDot } from './assets/icons/final/feather-folder-dot.js';
export { default as FeatherFolderDown } from './assets/icons/final/feather-folder-down.js';
export { default as FeatherFolderEdit } from './assets/icons/final/feather-folder-edit.js';
export { default as FeatherFolderGit2 } from './assets/icons/final/feather-folder-git-2.js';
export { default as FeatherFolderGit } from './assets/icons/final/feather-folder-git.js';
export { default as FeatherFolderHeart } from './assets/icons/final/feather-folder-heart.js';
export { default as FeatherFolderInput } from './assets/icons/final/feather-folder-input.js';
export { default as FeatherFolderKanban } from './assets/icons/final/feather-folder-kanban.js';
export { default as FeatherFolderKey } from './assets/icons/final/feather-folder-key.js';
export { default as FeatherFolderLock } from './assets/icons/final/feather-folder-lock.js';
export { default as FeatherFolderMinus } from './assets/icons/final/feather-folder-minus.js';
export { default as FeatherFolderOpenDot } from './assets/icons/final/feather-folder-open-dot.js';
export { default as FeatherFolderOpen } from './assets/icons/final/feather-folder-open.js';
export { default as FeatherFolderOutput } from './assets/icons/final/feather-folder-output.js';
export { default as FeatherFolderPen } from './assets/icons/final/feather-folder-pen.js';
export { default as FeatherFolderPlus } from './assets/icons/final/feather-folder-plus.js';
export { default as FeatherFolderRoot } from './assets/icons/final/feather-folder-root.js';
export { default as FeatherFolderSearch2 } from './assets/icons/final/feather-folder-search-2.js';
export { default as FeatherFolderSearch } from './assets/icons/final/feather-folder-search.js';
export { default as FeatherFolderSymlink } from './assets/icons/final/feather-folder-symlink.js';
export { default as FeatherFolderSync } from './assets/icons/final/feather-folder-sync.js';
export { default as FeatherFolderTree } from './assets/icons/final/feather-folder-tree.js';
export { default as FeatherFolderUp } from './assets/icons/final/feather-folder-up.js';
export { default as FeatherFolderX } from './assets/icons/final/feather-folder-x.js';
export { default as FeatherFolder } from './assets/icons/final/feather-folder.js';
export { default as FeatherFolders } from './assets/icons/final/feather-folders.js';
export { default as FeatherFootprints } from './assets/icons/final/feather-footprints.js';
export { default as FeatherForklift } from './assets/icons/final/feather-forklift.js';
export { default as FeatherFormInput } from './assets/icons/final/feather-form-input.js';
export { default as FeatherForward } from './assets/icons/final/feather-forward.js';
export { default as FeatherFrame } from './assets/icons/final/feather-frame.js';
export { default as FeatherFramer } from './assets/icons/final/feather-framer.js';
export { default as FeatherFrown } from './assets/icons/final/feather-frown.js';
export { default as FeatherFuel } from './assets/icons/final/feather-fuel.js';
export { default as FeatherFullscreen } from './assets/icons/final/feather-fullscreen.js';
export { default as FeatherFunctionSquare } from './assets/icons/final/feather-function-square.js';
export { default as FeatherFunnelPlus } from './assets/icons/final/feather-funnel-plus.js';
export { default as FeatherFunnelX } from './assets/icons/final/feather-funnel-x.js';
export { default as FeatherFunnel } from './assets/icons/final/feather-funnel.js';
export { default as FeatherGalleryHorizontalEnd } from './assets/icons/final/feather-gallery-horizontal-end.js';
export { default as FeatherGalleryHorizontal } from './assets/icons/final/feather-gallery-horizontal.js';
export { default as FeatherGalleryThumbnails } from './assets/icons/final/feather-gallery-thumbnails.js';
export { default as FeatherGalleryVerticalEnd } from './assets/icons/final/feather-gallery-vertical-end.js';
export { default as FeatherGalleryVertical } from './assets/icons/final/feather-gallery-vertical.js';
export { default as FeatherGamepad2 } from './assets/icons/final/feather-gamepad-2.js';
export { default as FeatherGamepad } from './assets/icons/final/feather-gamepad.js';
export { default as FeatherGanttChartSquare } from './assets/icons/final/feather-gantt-chart-square.js';
export { default as FeatherGanttChart } from './assets/icons/final/feather-gantt-chart.js';
export { default as FeatherGaugeCircle } from './assets/icons/final/feather-gauge-circle.js';
export { default as FeatherGauge } from './assets/icons/final/feather-gauge.js';
export { default as FeatherGavel } from './assets/icons/final/feather-gavel.js';
export { default as FeatherGem } from './assets/icons/final/feather-gem.js';
export { default as FeatherGeorgianLari } from './assets/icons/final/feather-georgian-lari.js';
export { default as FeatherGhost } from './assets/icons/final/feather-ghost.js';
export { default as FeatherGift } from './assets/icons/final/feather-gift.js';
export { default as FeatherGitBranchPlus } from './assets/icons/final/feather-git-branch-plus.js';
export { default as FeatherGitBranch } from './assets/icons/final/feather-git-branch.js';
export { default as FeatherGitCommitHorizontal } from './assets/icons/final/feather-git-commit-horizontal.js';
export { default as FeatherGitCommitVertical } from './assets/icons/final/feather-git-commit-vertical.js';
export { default as FeatherGitCommit } from './assets/icons/final/feather-git-commit.js';
export { default as FeatherGitCompareArrows } from './assets/icons/final/feather-git-compare-arrows.js';
export { default as FeatherGitCompare } from './assets/icons/final/feather-git-compare.js';
export { default as FeatherGitFork } from './assets/icons/final/feather-git-fork.js';
export { default as FeatherGitGraph } from './assets/icons/final/feather-git-graph.js';
export { default as FeatherGitMerge } from './assets/icons/final/feather-git-merge.js';
export { default as FeatherGitPullRequestArrow } from './assets/icons/final/feather-git-pull-request-arrow.js';
export { default as FeatherGitPullRequestClosed } from './assets/icons/final/feather-git-pull-request-closed.js';
export { default as FeatherGitPullRequestCreateArrow } from './assets/icons/final/feather-git-pull-request-create-arrow.js';
export { default as FeatherGitPullRequestCreate } from './assets/icons/final/feather-git-pull-request-create.js';
export { default as FeatherGitPullRequestDraft } from './assets/icons/final/feather-git-pull-request-draft.js';
export { default as FeatherGitPullRequest } from './assets/icons/final/feather-git-pull-request.js';
export { default as FeatherGithub } from './assets/icons/final/feather-github.js';
export { default as FeatherGitlab } from './assets/icons/final/feather-gitlab.js';
export { default as FeatherGlassWater } from './assets/icons/final/feather-glass-water.js';
export { default as FeatherGlasses } from './assets/icons/final/feather-glasses.js';
export { default as FeatherGlobe2 } from './assets/icons/final/feather-globe-2.js';
export { default as FeatherGlobeLock } from './assets/icons/final/feather-globe-lock.js';
export { default as FeatherGlobe } from './assets/icons/final/feather-globe.js';
export { default as FeatherGoal } from './assets/icons/final/feather-goal.js';
export { default as FeatherGpu } from './assets/icons/final/feather-gpu.js';
export { default as FeatherGrab } from './assets/icons/final/feather-grab.js';
export { default as FeatherGraduationCap } from './assets/icons/final/feather-graduation-cap.js';
export { default as FeatherGrape } from './assets/icons/final/feather-grape.js';
export { default as FeatherGrid2X2Check } from './assets/icons/final/feather-grid-2x2-check.js';
export { default as FeatherGrid2X2Plus } from './assets/icons/final/feather-grid-2x2-plus.js';
export { default as FeatherGrid2X2X } from './assets/icons/final/feather-grid-2x2-x.js';
export { default as FeatherGrid2X2 } from './assets/icons/final/feather-grid-2x2.js';
export { default as FeatherGrid3X2 } from './assets/icons/final/feather-grid-3x2.js';
export { default as FeatherGrid3X3 } from './assets/icons/final/feather-grid-3x3.js';
export { default as FeatherGrid } from './assets/icons/final/feather-grid.js';
export { default as FeatherGripHorizontal } from './assets/icons/final/feather-grip-horizontal.js';
export { default as FeatherGripVertical } from './assets/icons/final/feather-grip-vertical.js';
export { default as FeatherGrip } from './assets/icons/final/feather-grip.js';
export { default as FeatherGroup } from './assets/icons/final/feather-group.js';
export { default as FeatherGuitar } from './assets/icons/final/feather-guitar.js';
export { default as FeatherHam } from './assets/icons/final/feather-ham.js';
export { default as FeatherHamburger } from './assets/icons/final/feather-hamburger.js';
export { default as FeatherHammer } from './assets/icons/final/feather-hammer.js';
export { default as FeatherHandCoins } from './assets/icons/final/feather-hand-coins.js';
export { default as FeatherHandHeart } from './assets/icons/final/feather-hand-heart.js';
export { default as FeatherHandHelping } from './assets/icons/final/feather-hand-helping.js';
export { default as FeatherHandMetal } from './assets/icons/final/feather-hand-metal.js';
export { default as FeatherHandPlatter } from './assets/icons/final/feather-hand-platter.js';
export { default as FeatherHand } from './assets/icons/final/feather-hand.js';
export { default as FeatherHandshake } from './assets/icons/final/feather-handshake.js';
export { default as FeatherHardDriveDownload } from './assets/icons/final/feather-hard-drive-download.js';
export { default as FeatherHardDriveUpload } from './assets/icons/final/feather-hard-drive-upload.js';
export { default as FeatherHardDrive } from './assets/icons/final/feather-hard-drive.js';
export { default as FeatherHardHat } from './assets/icons/final/feather-hard-hat.js';
export { default as FeatherHash } from './assets/icons/final/feather-hash.js';
export { default as FeatherHaze } from './assets/icons/final/feather-haze.js';
export { default as FeatherHdmiPort } from './assets/icons/final/feather-hdmi-port.js';
export { default as FeatherHeading1 } from './assets/icons/final/feather-heading-1.js';
export { default as FeatherHeading2 } from './assets/icons/final/feather-heading-2.js';
export { default as FeatherHeading3 } from './assets/icons/final/feather-heading-3.js';
export { default as FeatherHeading4 } from './assets/icons/final/feather-heading-4.js';
export { default as FeatherHeading5 } from './assets/icons/final/feather-heading-5.js';
export { default as FeatherHeading6 } from './assets/icons/final/feather-heading-6.js';
export { default as FeatherHeading } from './assets/icons/final/feather-heading.js';
export { default as FeatherHeadphoneOff } from './assets/icons/final/feather-headphone-off.js';
export { default as FeatherHeadphones } from './assets/icons/final/feather-headphones.js';
export { default as FeatherHeadset } from './assets/icons/final/feather-headset.js';
export { default as FeatherHeartCrack } from './assets/icons/final/feather-heart-crack.js';
export { default as FeatherHeartHandshake } from './assets/icons/final/feather-heart-handshake.js';
export { default as FeatherHeartMinus } from './assets/icons/final/feather-heart-minus.js';
export { default as FeatherHeartOff } from './assets/icons/final/feather-heart-off.js';
export { default as FeatherHeartPlus } from './assets/icons/final/feather-heart-plus.js';
export { default as FeatherHeartPulse } from './assets/icons/final/feather-heart-pulse.js';
export { default as FeatherHeart } from './assets/icons/final/feather-heart.js';
export { default as FeatherHeater } from './assets/icons/final/feather-heater.js';
export { default as FeatherHelpCircle } from './assets/icons/final/feather-help-circle.js';
export { default as FeatherHelpingHand } from './assets/icons/final/feather-helping-hand.js';
export { default as FeatherHexagon } from './assets/icons/final/feather-hexagon.js';
export { default as FeatherHighlighter } from './assets/icons/final/feather-highlighter.js';
export { default as FeatherHistory } from './assets/icons/final/feather-history.js';
export { default as FeatherHome } from './assets/icons/final/feather-home.js';
export { default as FeatherHopOff } from './assets/icons/final/feather-hop-off.js';
export { default as FeatherHop } from './assets/icons/final/feather-hop.js';
export { default as FeatherHospital } from './assets/icons/final/feather-hospital.js';
export { default as FeatherHotel } from './assets/icons/final/feather-hotel.js';
export { default as FeatherHourglass } from './assets/icons/final/feather-hourglass.js';
export { default as FeatherHousePlug } from './assets/icons/final/feather-house-plug.js';
export { default as FeatherHousePlus } from './assets/icons/final/feather-house-plus.js';
export { default as FeatherHouseWifi } from './assets/icons/final/feather-house-wifi.js';
export { default as FeatherHouse } from './assets/icons/final/feather-house.js';
export { default as FeatherIceCream2 } from './assets/icons/final/feather-ice-cream-2.js';
export { default as FeatherIceCreamBowl } from './assets/icons/final/feather-ice-cream-bowl.js';
export { default as FeatherIceCream } from './assets/icons/final/feather-ice-cream.js';
export { default as FeatherIdCardLanyard } from './assets/icons/final/feather-id-card-lanyard.js';
export { default as FeatherIdCard } from './assets/icons/final/feather-id-card.js';
export { default as FeatherImageDown } from './assets/icons/final/feather-image-down.js';
export { default as FeatherImageMinus } from './assets/icons/final/feather-image-minus.js';
export { default as FeatherImageOff } from './assets/icons/final/feather-image-off.js';
export { default as FeatherImagePlay } from './assets/icons/final/feather-image-play.js';
export { default as FeatherImagePlus } from './assets/icons/final/feather-image-plus.js';
export { default as FeatherImageUp } from './assets/icons/final/feather-image-up.js';
export { default as FeatherImageUpscale } from './assets/icons/final/feather-image-upscale.js';
export { default as FeatherImage } from './assets/icons/final/feather-image.js';
export { default as FeatherImages } from './assets/icons/final/feather-images.js';
export { default as FeatherImport } from './assets/icons/final/feather-import.js';
export { default as FeatherInbox } from './assets/icons/final/feather-inbox.js';
export { default as FeatherIndentDecrease } from './assets/icons/final/feather-indent-decrease.js';
export { default as FeatherIndentIncrease } from './assets/icons/final/feather-indent-increase.js';
export { default as FeatherIndent } from './assets/icons/final/feather-indent.js';
export { default as FeatherIndianRupee } from './assets/icons/final/feather-indian-rupee.js';
export { default as FeatherInfinity } from './assets/icons/final/feather-infinity.js';
export { default as FeatherInfo } from './assets/icons/final/feather-info.js';
export { default as FeatherInspect } from './assets/icons/final/feather-inspect.js';
export { default as FeatherInspectionPanel } from './assets/icons/final/feather-inspection-panel.js';
export { default as FeatherInstagram } from './assets/icons/final/feather-instagram.js';
export { default as FeatherItalic } from './assets/icons/final/feather-italic.js';
export { default as FeatherIterationCcw } from './assets/icons/final/feather-iteration-ccw.js';
export { default as FeatherIterationCw } from './assets/icons/final/feather-iteration-cw.js';
export { default as FeatherJapaneseYen } from './assets/icons/final/feather-japanese-yen.js';
export { default as FeatherJoystick } from './assets/icons/final/feather-joystick.js';
export { default as FeatherKanbanSquareDashed } from './assets/icons/final/feather-kanban-square-dashed.js';
export { default as FeatherKanbanSquare } from './assets/icons/final/feather-kanban-square.js';
export { default as FeatherKanban } from './assets/icons/final/feather-kanban.js';
export { default as FeatherKeyRound } from './assets/icons/final/feather-key-round.js';
export { default as FeatherKeySquare } from './assets/icons/final/feather-key-square.js';
export { default as FeatherKey } from './assets/icons/final/feather-key.js';
export { default as FeatherKeyboardMusic } from './assets/icons/final/feather-keyboard-music.js';
export { default as FeatherKeyboardOff } from './assets/icons/final/feather-keyboard-off.js';
export { default as FeatherKeyboard } from './assets/icons/final/feather-keyboard.js';
export { default as FeatherLampCeiling } from './assets/icons/final/feather-lamp-ceiling.js';
export { default as FeatherLampDesk } from './assets/icons/final/feather-lamp-desk.js';
export { default as FeatherLampFloor } from './assets/icons/final/feather-lamp-floor.js';
export { default as FeatherLampWallDown } from './assets/icons/final/feather-lamp-wall-down.js';
export { default as FeatherLampWallUp } from './assets/icons/final/feather-lamp-wall-up.js';
export { default as FeatherLamp } from './assets/icons/final/feather-lamp.js';
export { default as FeatherLandPlot } from './assets/icons/final/feather-land-plot.js';
export { default as FeatherLandmark } from './assets/icons/final/feather-landmark.js';
export { default as FeatherLanguages } from './assets/icons/final/feather-languages.js';
export { default as FeatherLaptop2 } from './assets/icons/final/feather-laptop-2.js';
export { default as FeatherLaptopMinimalCheck } from './assets/icons/final/feather-laptop-minimal-check.js';
export { default as FeatherLaptop } from './assets/icons/final/feather-laptop.js';
export { default as FeatherLassoSelect } from './assets/icons/final/feather-lasso-select.js';
export { default as FeatherLasso } from './assets/icons/final/feather-lasso.js';
export { default as FeatherLaugh } from './assets/icons/final/feather-laugh.js';
export { default as FeatherLayers2 } from './assets/icons/final/feather-layers-2.js';
export { default as FeatherLayers3 } from './assets/icons/final/feather-layers-3.js';
export { default as FeatherLayers } from './assets/icons/final/feather-layers.js';
export { default as FeatherLayoutDashboard } from './assets/icons/final/feather-layout-dashboard.js';
export { default as FeatherLayoutGrid } from './assets/icons/final/feather-layout-grid.js';
export { default as FeatherLayoutList } from './assets/icons/final/feather-layout-list.js';
export { default as FeatherLayoutPanelLeft } from './assets/icons/final/feather-layout-panel-left.js';
export { default as FeatherLayoutPanelTop } from './assets/icons/final/feather-layout-panel-top.js';
export { default as FeatherLayoutTemplate } from './assets/icons/final/feather-layout-template.js';
export { default as FeatherLayout } from './assets/icons/final/feather-layout.js';
export { default as FeatherLeaf } from './assets/icons/final/feather-leaf.js';
export { default as FeatherLeafyGreen } from './assets/icons/final/feather-leafy-green.js';
export { default as FeatherLectern } from './assets/icons/final/feather-lectern.js';
export { default as FeatherLetterSpacing } from './assets/icons/final/feather-letter-spacing.js';
export { default as FeatherLetterText } from './assets/icons/final/feather-letter-text.js';
export { default as FeatherLibraryBig } from './assets/icons/final/feather-library-big.js';
export { default as FeatherLibrarySquare } from './assets/icons/final/feather-library-square.js';
export { default as FeatherLibrary } from './assets/icons/final/feather-library.js';
export { default as FeatherLifeBuoy } from './assets/icons/final/feather-life-buoy.js';
export { default as FeatherLigature } from './assets/icons/final/feather-ligature.js';
export { default as FeatherLightbulbOff } from './assets/icons/final/feather-lightbulb-off.js';
export { default as FeatherLightbulb } from './assets/icons/final/feather-lightbulb.js';
export { default as FeatherLineChart } from './assets/icons/final/feather-line-chart.js';
export { default as FeatherLineHeight } from './assets/icons/final/feather-line-height.js';
export { default as FeatherLineSquiggle } from './assets/icons/final/feather-line-squiggle.js';
export { default as FeatherLink2Off } from './assets/icons/final/feather-link-2-off.js';
export { default as FeatherLink2 } from './assets/icons/final/feather-link-2.js';
export { default as FeatherLink } from './assets/icons/final/feather-link.js';
export { default as FeatherLinkedin } from './assets/icons/final/feather-linkedin.js';
export { default as FeatherListCheck } from './assets/icons/final/feather-list-check.js';
export { default as FeatherListChecks } from './assets/icons/final/feather-list-checks.js';
export { default as FeatherListCollapse } from './assets/icons/final/feather-list-collapse.js';
export { default as FeatherListEnd } from './assets/icons/final/feather-list-end.js';
export { default as FeatherListFilterPlus } from './assets/icons/final/feather-list-filter-plus.js';
export { default as FeatherListFilter } from './assets/icons/final/feather-list-filter.js';
export { default as FeatherListMinus } from './assets/icons/final/feather-list-minus.js';
export { default as FeatherListMusic } from './assets/icons/final/feather-list-music.js';
export { default as FeatherListOrdered } from './assets/icons/final/feather-list-ordered.js';
export { default as FeatherListPlus } from './assets/icons/final/feather-list-plus.js';
export { default as FeatherListRestart } from './assets/icons/final/feather-list-restart.js';
export { default as FeatherListStart } from './assets/icons/final/feather-list-start.js';
export { default as FeatherListTodo } from './assets/icons/final/feather-list-todo.js';
export { default as FeatherListTree } from './assets/icons/final/feather-list-tree.js';
export { default as FeatherListVideo } from './assets/icons/final/feather-list-video.js';
export { default as FeatherListX } from './assets/icons/final/feather-list-x.js';
export { default as FeatherList } from './assets/icons/final/feather-list.js';
export { default as FeatherLoader2 } from './assets/icons/final/feather-loader-2.js';
export { default as FeatherLoaderPinwheel } from './assets/icons/final/feather-loader-pinwheel.js';
export { default as FeatherLoader } from './assets/icons/final/feather-loader.js';
export { default as FeatherLocateFixed } from './assets/icons/final/feather-locate-fixed.js';
export { default as FeatherLocateOff } from './assets/icons/final/feather-locate-off.js';
export { default as FeatherLocate } from './assets/icons/final/feather-locate.js';
export { default as FeatherLocationEdit } from './assets/icons/final/feather-location-edit.js';
export { default as FeatherLockKeyholeOpen } from './assets/icons/final/feather-lock-keyhole-open.js';
export { default as FeatherLockKeyhole } from './assets/icons/final/feather-lock-keyhole.js';
export { default as FeatherLock } from './assets/icons/final/feather-lock.js';
export { default as FeatherLogIn } from './assets/icons/final/feather-log-in.js';
export { default as FeatherLogOut } from './assets/icons/final/feather-log-out.js';
export { default as FeatherLogs } from './assets/icons/final/feather-logs.js';
export { default as FeatherLollipop } from './assets/icons/final/feather-lollipop.js';
export { default as FeatherLuggage } from './assets/icons/final/feather-luggage.js';
export { default as FeatherMSquare } from './assets/icons/final/feather-m-square.js';
export { default as FeatherMagnet } from './assets/icons/final/feather-magnet.js';
export { default as FeatherMailCheck } from './assets/icons/final/feather-mail-check.js';
export { default as FeatherMailMinus } from './assets/icons/final/feather-mail-minus.js';
export { default as FeatherMailOpen } from './assets/icons/final/feather-mail-open.js';
export { default as FeatherMailPlus } from './assets/icons/final/feather-mail-plus.js';
export { default as FeatherMailQuestion } from './assets/icons/final/feather-mail-question.js';
export { default as FeatherMailSearch } from './assets/icons/final/feather-mail-search.js';
export { default as FeatherMailWarning } from './assets/icons/final/feather-mail-warning.js';
export { default as FeatherMailX } from './assets/icons/final/feather-mail-x.js';
export { default as FeatherMail } from './assets/icons/final/feather-mail.js';
export { default as FeatherMailbox } from './assets/icons/final/feather-mailbox.js';
export { default as FeatherMails } from './assets/icons/final/feather-mails.js';
export { default as FeatherMapPinCheckInside } from './assets/icons/final/feather-map-pin-check-inside.js';
export { default as FeatherMapPinCheck } from './assets/icons/final/feather-map-pin-check.js';
export { default as FeatherMapPinHouse } from './assets/icons/final/feather-map-pin-house.js';
export { default as FeatherMapPinMinusInside } from './assets/icons/final/feather-map-pin-minus-inside.js';
export { default as FeatherMapPinMinus } from './assets/icons/final/feather-map-pin-minus.js';
export { default as FeatherMapPinOff } from './assets/icons/final/feather-map-pin-off.js';
export { default as FeatherMapPinPlusInside } from './assets/icons/final/feather-map-pin-plus-inside.js';
export { default as FeatherMapPinPlus } from './assets/icons/final/feather-map-pin-plus.js';
export { default as FeatherMapPinXInside } from './assets/icons/final/feather-map-pin-x-inside.js';
export { default as FeatherMapPinX } from './assets/icons/final/feather-map-pin-x.js';
export { default as FeatherMapPin } from './assets/icons/final/feather-map-pin.js';
export { default as FeatherMapPinned } from './assets/icons/final/feather-map-pinned.js';
export { default as FeatherMapPlus } from './assets/icons/final/feather-map-plus.js';
export { default as FeatherMap } from './assets/icons/final/feather-map.js';
export { default as FeatherMarsStroke } from './assets/icons/final/feather-mars-stroke.js';
export { default as FeatherMars } from './assets/icons/final/feather-mars.js';
export { default as FeatherMartini } from './assets/icons/final/feather-martini.js';
export { default as FeatherMaximize2 } from './assets/icons/final/feather-maximize-2.js';
export { default as FeatherMaximize } from './assets/icons/final/feather-maximize.js';
export { default as FeatherMedal } from './assets/icons/final/feather-medal.js';
export { default as FeatherMegaphoneOff } from './assets/icons/final/feather-megaphone-off.js';
export { default as FeatherMegaphone } from './assets/icons/final/feather-megaphone.js';
export { default as FeatherMeh } from './assets/icons/final/feather-meh.js';
export { default as FeatherMemoryStick } from './assets/icons/final/feather-memory-stick.js';
export { default as FeatherMenuSquare } from './assets/icons/final/feather-menu-square.js';
export { default as FeatherMenu } from './assets/icons/final/feather-menu.js';
export { default as FeatherMerge } from './assets/icons/final/feather-merge.js';
export { default as FeatherMessageCircleCode } from './assets/icons/final/feather-message-circle-code.js';
export { default as FeatherMessageCircleDashed } from './assets/icons/final/feather-message-circle-dashed.js';
export { default as FeatherMessageCircleHeart } from './assets/icons/final/feather-message-circle-heart.js';
export { default as FeatherMessageCircleMore } from './assets/icons/final/feather-message-circle-more.js';
export { default as FeatherMessageCircleOff } from './assets/icons/final/feather-message-circle-off.js';
export { default as FeatherMessageCirclePlus } from './assets/icons/final/feather-message-circle-plus.js';
export { default as FeatherMessageCircleQuestionMark } from './assets/icons/final/feather-message-circle-question-mark.js';
export { default as FeatherMessageCircleReply } from './assets/icons/final/feather-message-circle-reply.js';
export { default as FeatherMessageCircleWarning } from './assets/icons/final/feather-message-circle-warning.js';
export { default as FeatherMessageCircleX } from './assets/icons/final/feather-message-circle-x.js';
export { default as FeatherMessageCircle } from './assets/icons/final/feather-message-circle.js';
export { default as FeatherMessageSquareCode } from './assets/icons/final/feather-message-square-code.js';
export { default as FeatherMessageSquareDashed } from './assets/icons/final/feather-message-square-dashed.js';
export { default as FeatherMessageSquareDiff } from './assets/icons/final/feather-message-square-diff.js';
export { default as FeatherMessageSquareDot } from './assets/icons/final/feather-message-square-dot.js';
export { default as FeatherMessageSquareHeart } from './assets/icons/final/feather-message-square-heart.js';
export { default as FeatherMessageSquareLock } from './assets/icons/final/feather-message-square-lock.js';
export { default as FeatherMessageSquareMore } from './assets/icons/final/feather-message-square-more.js';
export { default as FeatherMessageSquareOff } from './assets/icons/final/feather-message-square-off.js';
export { default as FeatherMessageSquarePlus } from './assets/icons/final/feather-message-square-plus.js';
export { default as FeatherMessageSquareQuote } from './assets/icons/final/feather-message-square-quote.js';
export { default as FeatherMessageSquareReply } from './assets/icons/final/feather-message-square-reply.js';
export { default as FeatherMessageSquareShare } from './assets/icons/final/feather-message-square-share.js';
export { default as FeatherMessageSquareText } from './assets/icons/final/feather-message-square-text.js';
export { default as FeatherMessageSquareWarning } from './assets/icons/final/feather-message-square-warning.js';
export { default as FeatherMessageSquareX } from './assets/icons/final/feather-message-square-x.js';
export { default as FeatherMessageSquare } from './assets/icons/final/feather-message-square.js';
export { default as FeatherMessagesSquare } from './assets/icons/final/feather-messages-square.js';
export { default as FeatherMic2 } from './assets/icons/final/feather-mic-2.js';
export { default as FeatherMicOff } from './assets/icons/final/feather-mic-off.js';
export { default as FeatherMicVocal } from './assets/icons/final/feather-mic-vocal.js';
export { default as FeatherMic } from './assets/icons/final/feather-mic.js';
export { default as FeatherMicrochip } from './assets/icons/final/feather-microchip.js';
export { default as FeatherMicroscope } from './assets/icons/final/feather-microscope.js';
export { default as FeatherMicrowave } from './assets/icons/final/feather-microwave.js';
export { default as FeatherMilestone } from './assets/icons/final/feather-milestone.js';
export { default as FeatherMilkOff } from './assets/icons/final/feather-milk-off.js';
export { default as FeatherMilk } from './assets/icons/final/feather-milk.js';
export { default as FeatherMinimize2 } from './assets/icons/final/feather-minimize-2.js';
export { default as FeatherMinimize } from './assets/icons/final/feather-minimize.js';
export { default as FeatherMinusCircle } from './assets/icons/final/feather-minus-circle.js';
export { default as FeatherMinusSquare } from './assets/icons/final/feather-minus-square.js';
export { default as FeatherMinus } from './assets/icons/final/feather-minus.js';
export { default as FeatherMonitorCheck } from './assets/icons/final/feather-monitor-check.js';
export { default as FeatherMonitorCog } from './assets/icons/final/feather-monitor-cog.js';
export { default as FeatherMonitorDot } from './assets/icons/final/feather-monitor-dot.js';
export { default as FeatherMonitorDown } from './assets/icons/final/feather-monitor-down.js';
export { default as FeatherMonitorOff } from './assets/icons/final/feather-monitor-off.js';
export { default as FeatherMonitorPause } from './assets/icons/final/feather-monitor-pause.js';
export { default as FeatherMonitorPlay } from './assets/icons/final/feather-monitor-play.js';
export { default as FeatherMonitorSmartphone } from './assets/icons/final/feather-monitor-smartphone.js';
export { default as FeatherMonitorSpeaker } from './assets/icons/final/feather-monitor-speaker.js';
export { default as FeatherMonitorStop } from './assets/icons/final/feather-monitor-stop.js';
export { default as FeatherMonitorUp } from './assets/icons/final/feather-monitor-up.js';
export { default as FeatherMonitorX } from './assets/icons/final/feather-monitor-x.js';
export { default as FeatherMonitor } from './assets/icons/final/feather-monitor.js';
export { default as FeatherMoonStar } from './assets/icons/final/feather-moon-star.js';
export { default as FeatherMoon } from './assets/icons/final/feather-moon.js';
export { default as FeatherMoreHorizontal } from './assets/icons/final/feather-more-horizontal.js';
export { default as FeatherMoreVertical } from './assets/icons/final/feather-more-vertical.js';
export { default as FeatherMountainSnow } from './assets/icons/final/feather-mountain-snow.js';
export { default as FeatherMountain } from './assets/icons/final/feather-mountain.js';
export { default as FeatherMouseOff } from './assets/icons/final/feather-mouse-off.js';
export { default as FeatherMousePointer2 } from './assets/icons/final/feather-mouse-pointer-2.js';
export { default as FeatherMousePointerBan } from './assets/icons/final/feather-mouse-pointer-ban.js';
export { default as FeatherMousePointerClick } from './assets/icons/final/feather-mouse-pointer-click.js';
export { default as FeatherMousePointerSquareDashed } from './assets/icons/final/feather-mouse-pointer-square-dashed.js';
export { default as FeatherMousePointerSquare } from './assets/icons/final/feather-mouse-pointer-square.js';
export { default as FeatherMousePointer } from './assets/icons/final/feather-mouse-pointer.js';
export { default as FeatherMouse } from './assets/icons/final/feather-mouse.js';
export { default as FeatherMove3D } from './assets/icons/final/feather-move-3d.js';
export { default as FeatherMoveDiagonal2 } from './assets/icons/final/feather-move-diagonal-2.js';
export { default as FeatherMoveDiagonal } from './assets/icons/final/feather-move-diagonal.js';
export { default as FeatherMoveDownLeft } from './assets/icons/final/feather-move-down-left.js';
export { default as FeatherMoveDownRight } from './assets/icons/final/feather-move-down-right.js';
export { default as FeatherMoveDown } from './assets/icons/final/feather-move-down.js';
export { default as FeatherMoveHorizontal } from './assets/icons/final/feather-move-horizontal.js';
export { default as FeatherMoveLeft } from './assets/icons/final/feather-move-left.js';
export { default as FeatherMoveRight } from './assets/icons/final/feather-move-right.js';
export { default as FeatherMoveUpLeft } from './assets/icons/final/feather-move-up-left.js';
export { default as FeatherMoveUpRight } from './assets/icons/final/feather-move-up-right.js';
export { default as FeatherMoveUp } from './assets/icons/final/feather-move-up.js';
export { default as FeatherMoveVertical } from './assets/icons/final/feather-move-vertical.js';
export { default as FeatherMove } from './assets/icons/final/feather-move.js';
export { default as FeatherMusic2 } from './assets/icons/final/feather-music-2.js';
export { default as FeatherMusic3 } from './assets/icons/final/feather-music-3.js';
export { default as FeatherMusic4 } from './assets/icons/final/feather-music-4.js';
export { default as FeatherMusic } from './assets/icons/final/feather-music.js';
export { default as FeatherNavigation2Off } from './assets/icons/final/feather-navigation-2-off.js';
export { default as FeatherNavigation2 } from './assets/icons/final/feather-navigation-2.js';
export { default as FeatherNavigationOff } from './assets/icons/final/feather-navigation-off.js';
export { default as FeatherNavigation } from './assets/icons/final/feather-navigation.js';
export { default as FeatherNetwork } from './assets/icons/final/feather-network.js';
export { default as FeatherNewspaper } from './assets/icons/final/feather-newspaper.js';
export { default as FeatherNfc } from './assets/icons/final/feather-nfc.js';
export { default as FeatherNonBinary } from './assets/icons/final/feather-non-binary.js';
export { default as FeatherNotebookPen } from './assets/icons/final/feather-notebook-pen.js';
export { default as FeatherNotebookTabs } from './assets/icons/final/feather-notebook-tabs.js';
export { default as FeatherNotebookText } from './assets/icons/final/feather-notebook-text.js';
export { default as FeatherNotebook } from './assets/icons/final/feather-notebook.js';
export { default as FeatherNotepadTextDashed } from './assets/icons/final/feather-notepad-text-dashed.js';
export { default as FeatherNotepadText } from './assets/icons/final/feather-notepad-text.js';
export { default as FeatherNutOff } from './assets/icons/final/feather-nut-off.js';
export { default as FeatherNut } from './assets/icons/final/feather-nut.js';
export { default as FeatherOctagonAlert } from './assets/icons/final/feather-octagon-alert.js';
export { default as FeatherOctagonMinus } from './assets/icons/final/feather-octagon-minus.js';
export { default as FeatherOctagonPause } from './assets/icons/final/feather-octagon-pause.js';
export { default as FeatherOctagonX } from './assets/icons/final/feather-octagon-x.js';
export { default as FeatherOctagon } from './assets/icons/final/feather-octagon.js';
export { default as FeatherOmega } from './assets/icons/final/feather-omega.js';
export { default as FeatherOption } from './assets/icons/final/feather-option.js';
export { default as FeatherOrbit } from './assets/icons/final/feather-orbit.js';
export { default as FeatherOrigami } from './assets/icons/final/feather-origami.js';
export { default as FeatherOutdent } from './assets/icons/final/feather-outdent.js';
export { default as FeatherPackage2 } from './assets/icons/final/feather-package-2.js';
export { default as FeatherPackageCheck } from './assets/icons/final/feather-package-check.js';
export { default as FeatherPackageMinus } from './assets/icons/final/feather-package-minus.js';
export { default as FeatherPackageOpen } from './assets/icons/final/feather-package-open.js';
export { default as FeatherPackagePlus } from './assets/icons/final/feather-package-plus.js';
export { default as FeatherPackageSearch } from './assets/icons/final/feather-package-search.js';
export { default as FeatherPackageX } from './assets/icons/final/feather-package-x.js';
export { default as FeatherPackage } from './assets/icons/final/feather-package.js';
export { default as FeatherPaintBucket } from './assets/icons/final/feather-paint-bucket.js';
export { default as FeatherPaintRoller } from './assets/icons/final/feather-paint-roller.js';
export { default as FeatherPaintbrush2 } from './assets/icons/final/feather-paintbrush-2.js';
export { default as FeatherPaintbrushVertical } from './assets/icons/final/feather-paintbrush-vertical.js';
export { default as FeatherPaintbrush } from './assets/icons/final/feather-paintbrush.js';
export { default as FeatherPalette } from './assets/icons/final/feather-palette.js';
export { default as FeatherPalmtree } from './assets/icons/final/feather-palmtree.js';
export { default as FeatherPanda } from './assets/icons/final/feather-panda.js';
export { default as FeatherPanelBottomClose } from './assets/icons/final/feather-panel-bottom-close.js';
export { default as FeatherPanelBottomInactive } from './assets/icons/final/feather-panel-bottom-inactive.js';
export { default as FeatherPanelBottomOpen } from './assets/icons/final/feather-panel-bottom-open.js';
export { default as FeatherPanelBottom } from './assets/icons/final/feather-panel-bottom.js';
export { default as FeatherPanelLeftClose } from './assets/icons/final/feather-panel-left-close.js';
export { default as FeatherPanelLeftInactive } from './assets/icons/final/feather-panel-left-inactive.js';
export { default as FeatherPanelLeftOpen } from './assets/icons/final/feather-panel-left-open.js';
export { default as FeatherPanelLeft } from './assets/icons/final/feather-panel-left.js';
export { default as FeatherPanelRightClose } from './assets/icons/final/feather-panel-right-close.js';
export { default as FeatherPanelRightInactive } from './assets/icons/final/feather-panel-right-inactive.js';
export { default as FeatherPanelRightOpen } from './assets/icons/final/feather-panel-right-open.js';
export { default as FeatherPanelRight } from './assets/icons/final/feather-panel-right.js';
export { default as FeatherPanelTopClose } from './assets/icons/final/feather-panel-top-close.js';
export { default as FeatherPanelTopInactive } from './assets/icons/final/feather-panel-top-inactive.js';
export { default as FeatherPanelTopOpen } from './assets/icons/final/feather-panel-top-open.js';
export { default as FeatherPanelTop } from './assets/icons/final/feather-panel-top.js';
export { default as FeatherPanelsLeftBottom } from './assets/icons/final/feather-panels-left-bottom.js';
export { default as FeatherPanelsRightBottom } from './assets/icons/final/feather-panels-right-bottom.js';
export { default as FeatherPanelsTopLeft } from './assets/icons/final/feather-panels-top-left.js';
export { default as FeatherPaperclip } from './assets/icons/final/feather-paperclip.js';
export { default as FeatherParentheses } from './assets/icons/final/feather-parentheses.js';
export { default as FeatherParkingCircleOff } from './assets/icons/final/feather-parking-circle-off.js';
export { default as FeatherParkingCircle } from './assets/icons/final/feather-parking-circle.js';
export { default as FeatherParkingMeter } from './assets/icons/final/feather-parking-meter.js';
export { default as FeatherParkingSquareOff } from './assets/icons/final/feather-parking-square-off.js';
export { default as FeatherParkingSquare } from './assets/icons/final/feather-parking-square.js';
export { default as FeatherPartyPopper } from './assets/icons/final/feather-party-popper.js';
export { default as FeatherPauseCircle } from './assets/icons/final/feather-pause-circle.js';
export { default as FeatherPauseOctagon } from './assets/icons/final/feather-pause-octagon.js';
export { default as FeatherPause } from './assets/icons/final/feather-pause.js';
export { default as FeatherPawPrint } from './assets/icons/final/feather-paw-print.js';
export { default as FeatherPcCase } from './assets/icons/final/feather-pc-case.js';
export { default as FeatherPenLine } from './assets/icons/final/feather-pen-line.js';
export { default as FeatherPenOff } from './assets/icons/final/feather-pen-off.js';
export { default as FeatherPenSquare } from './assets/icons/final/feather-pen-square.js';
export { default as FeatherPenTool } from './assets/icons/final/feather-pen-tool.js';
export { default as FeatherPen } from './assets/icons/final/feather-pen.js';
export { default as FeatherPencilLine } from './assets/icons/final/feather-pencil-line.js';
export { default as FeatherPencilOff } from './assets/icons/final/feather-pencil-off.js';
export { default as FeatherPencilRuler } from './assets/icons/final/feather-pencil-ruler.js';
export { default as FeatherPencil } from './assets/icons/final/feather-pencil.js';
export { default as FeatherPentagon } from './assets/icons/final/feather-pentagon.js';
export { default as FeatherPercentCircle } from './assets/icons/final/feather-percent-circle.js';
export { default as FeatherPercentDiamond } from './assets/icons/final/feather-percent-diamond.js';
export { default as FeatherPercentSquare } from './assets/icons/final/feather-percent-square.js';
export { default as FeatherPercent } from './assets/icons/final/feather-percent.js';
export { default as FeatherPersonStanding } from './assets/icons/final/feather-person-standing.js';
export { default as FeatherPhilippinePeso } from './assets/icons/final/feather-philippine-peso.js';
export { default as FeatherPhoneCall } from './assets/icons/final/feather-phone-call.js';
export { default as FeatherPhoneForwarded } from './assets/icons/final/feather-phone-forwarded.js';
export { default as FeatherPhoneIncoming } from './assets/icons/final/feather-phone-incoming.js';
export { default as FeatherPhoneMissed } from './assets/icons/final/feather-phone-missed.js';
export { default as FeatherPhoneOff } from './assets/icons/final/feather-phone-off.js';
export { default as FeatherPhoneOutgoing } from './assets/icons/final/feather-phone-outgoing.js';
export { default as FeatherPhone } from './assets/icons/final/feather-phone.js';
export { default as FeatherPiSquare } from './assets/icons/final/feather-pi-square.js';
export { default as FeatherPi } from './assets/icons/final/feather-pi.js';
export { default as FeatherPiano } from './assets/icons/final/feather-piano.js';
export { default as FeatherPickaxe } from './assets/icons/final/feather-pickaxe.js';
export { default as FeatherPictureInPicture2 } from './assets/icons/final/feather-picture-in-picture-2.js';
export { default as FeatherPictureInPicture } from './assets/icons/final/feather-picture-in-picture.js';
export { default as FeatherPieChart } from './assets/icons/final/feather-pie-chart.js';
export { default as FeatherPiggyBank } from './assets/icons/final/feather-piggy-bank.js';
export { default as FeatherPilcrowLeft } from './assets/icons/final/feather-pilcrow-left.js';
export { default as FeatherPilcrowRight } from './assets/icons/final/feather-pilcrow-right.js';
export { default as FeatherPilcrowSquare } from './assets/icons/final/feather-pilcrow-square.js';
export { default as FeatherPilcrow } from './assets/icons/final/feather-pilcrow.js';
export { default as FeatherPillBottle } from './assets/icons/final/feather-pill-bottle.js';
export { default as FeatherPill } from './assets/icons/final/feather-pill.js';
export { default as FeatherPinOff } from './assets/icons/final/feather-pin-off.js';
export { default as FeatherPin } from './assets/icons/final/feather-pin.js';
export { default as FeatherPipette } from './assets/icons/final/feather-pipette.js';
export { default as FeatherPizza } from './assets/icons/final/feather-pizza.js';
export { default as FeatherPlaneLanding } from './assets/icons/final/feather-plane-landing.js';
export { default as FeatherPlaneTakeoff } from './assets/icons/final/feather-plane-takeoff.js';
export { default as FeatherPlane } from './assets/icons/final/feather-plane.js';
export { default as FeatherPlayCircle } from './assets/icons/final/feather-play-circle.js';
export { default as FeatherPlaySquare } from './assets/icons/final/feather-play-square.js';
export { default as FeatherPlay } from './assets/icons/final/feather-play.js';
export { default as FeatherPlug2 } from './assets/icons/final/feather-plug-2.js';
export { default as FeatherPlugZap2 } from './assets/icons/final/feather-plug-zap-2.js';
export { default as FeatherPlugZap } from './assets/icons/final/feather-plug-zap.js';
export { default as FeatherPlug } from './assets/icons/final/feather-plug.js';
export { default as FeatherPlusCircle } from './assets/icons/final/feather-plus-circle.js';
export { default as FeatherPlusSquare } from './assets/icons/final/feather-plus-square.js';
export { default as FeatherPlus } from './assets/icons/final/feather-plus.js';
export { default as FeatherPocketKnife } from './assets/icons/final/feather-pocket-knife.js';
export { default as FeatherPocket } from './assets/icons/final/feather-pocket.js';
export { default as FeatherPodcast } from './assets/icons/final/feather-podcast.js';
export { default as FeatherPointerOff } from './assets/icons/final/feather-pointer-off.js';
export { default as FeatherPointer } from './assets/icons/final/feather-pointer.js';
export { default as FeatherPopcorn } from './assets/icons/final/feather-popcorn.js';
export { default as FeatherPopsicle } from './assets/icons/final/feather-popsicle.js';
export { default as FeatherPoundSterling } from './assets/icons/final/feather-pound-sterling.js';
export { default as FeatherPowerCircle } from './assets/icons/final/feather-power-circle.js';
export { default as FeatherPowerOff } from './assets/icons/final/feather-power-off.js';
export { default as FeatherPowerSquare } from './assets/icons/final/feather-power-square.js';
export { default as FeatherPower } from './assets/icons/final/feather-power.js';
export { default as FeatherPresentation } from './assets/icons/final/feather-presentation.js';
export { default as FeatherPrinterCheck } from './assets/icons/final/feather-printer-check.js';
export { default as FeatherPrinter } from './assets/icons/final/feather-printer.js';
export { default as FeatherProjector } from './assets/icons/final/feather-projector.js';
export { default as FeatherProportions } from './assets/icons/final/feather-proportions.js';
export { default as FeatherPuzzle } from './assets/icons/final/feather-puzzle.js';
export { default as FeatherPyramid } from './assets/icons/final/feather-pyramid.js';
export { default as FeatherQrCode } from './assets/icons/final/feather-qr-code.js';
export { default as FeatherQuote } from './assets/icons/final/feather-quote.js';
export { default as FeatherRabbit } from './assets/icons/final/feather-rabbit.js';
export { default as FeatherRadar } from './assets/icons/final/feather-radar.js';
export { default as FeatherRadiation } from './assets/icons/final/feather-radiation.js';
export { default as FeatherRadical } from './assets/icons/final/feather-radical.js';
export { default as FeatherRadioReceiver } from './assets/icons/final/feather-radio-receiver.js';
export { default as FeatherRadioTower } from './assets/icons/final/feather-radio-tower.js';
export { default as FeatherRadio } from './assets/icons/final/feather-radio.js';
export { default as FeatherRadius } from './assets/icons/final/feather-radius.js';
export { default as FeatherRailSymbol } from './assets/icons/final/feather-rail-symbol.js';
export { default as FeatherRainbow } from './assets/icons/final/feather-rainbow.js';
export { default as FeatherRat } from './assets/icons/final/feather-rat.js';
export { default as FeatherRatio } from './assets/icons/final/feather-ratio.js';
export { default as FeatherReceiptCent } from './assets/icons/final/feather-receipt-cent.js';
export { default as FeatherReceiptEuro } from './assets/icons/final/feather-receipt-euro.js';
export { default as FeatherReceiptIndianRupee } from './assets/icons/final/feather-receipt-indian-rupee.js';
export { default as FeatherReceiptJapaneseYen } from './assets/icons/final/feather-receipt-japanese-yen.js';
export { default as FeatherReceiptPoundSterling } from './assets/icons/final/feather-receipt-pound-sterling.js';
export { default as FeatherReceiptRussianRuble } from './assets/icons/final/feather-receipt-russian-ruble.js';
export { default as FeatherReceiptSwissFranc } from './assets/icons/final/feather-receipt-swiss-franc.js';
export { default as FeatherReceiptText } from './assets/icons/final/feather-receipt-text.js';
export { default as FeatherReceipt } from './assets/icons/final/feather-receipt.js';
export { default as FeatherRectangleCircle } from './assets/icons/final/feather-rectangle-circle.js';
export { default as FeatherRectangleGoggles } from './assets/icons/final/feather-rectangle-goggles.js';
export { default as FeatherRectangleHorizontal } from './assets/icons/final/feather-rectangle-horizontal.js';
export { default as FeatherRectangleVertical } from './assets/icons/final/feather-rectangle-vertical.js';
export { default as FeatherRecycle } from './assets/icons/final/feather-recycle.js';
export { default as FeatherRedo2 } from './assets/icons/final/feather-redo-2.js';
export { default as FeatherRedoDot } from './assets/icons/final/feather-redo-dot.js';
export { default as FeatherRedo } from './assets/icons/final/feather-redo.js';
export { default as FeatherRefreshCcwDot } from './assets/icons/final/feather-refresh-ccw-dot.js';
export { default as FeatherRefreshCcw } from './assets/icons/final/feather-refresh-ccw.js';
export { default as FeatherRefreshCwOff } from './assets/icons/final/feather-refresh-cw-off.js';
export { default as FeatherRefreshCw } from './assets/icons/final/feather-refresh-cw.js';
export { default as FeatherRefrigerator } from './assets/icons/final/feather-refrigerator.js';
export { default as FeatherRegex } from './assets/icons/final/feather-regex.js';
export { default as FeatherRemoveFormatting } from './assets/icons/final/feather-remove-formatting.js';
export { default as FeatherRepeat1 } from './assets/icons/final/feather-repeat-1.js';
export { default as FeatherRepeat2 } from './assets/icons/final/feather-repeat-2.js';
export { default as FeatherRepeat } from './assets/icons/final/feather-repeat.js';
export { default as FeatherReplaceAll } from './assets/icons/final/feather-replace-all.js';
export { default as FeatherReplace } from './assets/icons/final/feather-replace.js';
export { default as FeatherReplyAll } from './assets/icons/final/feather-reply-all.js';
export { default as FeatherReply } from './assets/icons/final/feather-reply.js';
export { default as FeatherRewind } from './assets/icons/final/feather-rewind.js';
export { default as FeatherRibbon } from './assets/icons/final/feather-ribbon.js';
export { default as FeatherRocket } from './assets/icons/final/feather-rocket.js';
export { default as FeatherRockingChair } from './assets/icons/final/feather-rocking-chair.js';
export { default as FeatherRollerCoaster } from './assets/icons/final/feather-roller-coaster.js';
export { default as FeatherRotate3D } from './assets/icons/final/feather-rotate-3d.js';
export { default as FeatherRotateCcwKey } from './assets/icons/final/feather-rotate-ccw-key.js';
export { default as FeatherRotateCcwSquare } from './assets/icons/final/feather-rotate-ccw-square.js';
export { default as FeatherRotateCcw } from './assets/icons/final/feather-rotate-ccw.js';
export { default as FeatherRotateCwSquare } from './assets/icons/final/feather-rotate-cw-square.js';
export { default as FeatherRotateCw } from './assets/icons/final/feather-rotate-cw.js';
export { default as FeatherRouteOff } from './assets/icons/final/feather-route-off.js';
export { default as FeatherRoute } from './assets/icons/final/feather-route.js';
export { default as FeatherRouter } from './assets/icons/final/feather-router.js';
export { default as FeatherRows2 } from './assets/icons/final/feather-rows-2.js';
export { default as FeatherRows3 } from './assets/icons/final/feather-rows-3.js';
export { default as FeatherRows4 } from './assets/icons/final/feather-rows-4.js';
export { default as FeatherRows } from './assets/icons/final/feather-rows.js';
export { default as FeatherRss } from './assets/icons/final/feather-rss.js';
export { default as FeatherRulerDimensionLine } from './assets/icons/final/feather-ruler-dimension-line.js';
export { default as FeatherRuler } from './assets/icons/final/feather-ruler.js';
export { default as FeatherRussianRuble } from './assets/icons/final/feather-russian-ruble.js';
export { default as FeatherSailboat } from './assets/icons/final/feather-sailboat.js';
export { default as FeatherSalad } from './assets/icons/final/feather-salad.js';
export { default as FeatherSandwich } from './assets/icons/final/feather-sandwich.js';
export { default as FeatherSatelliteDish } from './assets/icons/final/feather-satellite-dish.js';
export { default as FeatherSatellite } from './assets/icons/final/feather-satellite.js';
export { default as FeatherSaudiRiyal } from './assets/icons/final/feather-saudi-riyal.js';
export { default as FeatherSaveAll } from './assets/icons/final/feather-save-all.js';
export { default as FeatherSaveOff } from './assets/icons/final/feather-save-off.js';
export { default as FeatherSave } from './assets/icons/final/feather-save.js';
export { default as FeatherScale3D } from './assets/icons/final/feather-scale-3d.js';
export { default as FeatherScale } from './assets/icons/final/feather-scale.js';
export { default as FeatherScaling } from './assets/icons/final/feather-scaling.js';
export { default as FeatherScanBarcode } from './assets/icons/final/feather-scan-barcode.js';
export { default as FeatherScanEye } from './assets/icons/final/feather-scan-eye.js';
export { default as FeatherScanFace } from './assets/icons/final/feather-scan-face.js';
export { default as FeatherScanHeart } from './assets/icons/final/feather-scan-heart.js';
export { default as FeatherScanLine } from './assets/icons/final/feather-scan-line.js';
export { default as FeatherScanQrCode } from './assets/icons/final/feather-scan-qr-code.js';
export { default as FeatherScanSearch } from './assets/icons/final/feather-scan-search.js';
export { default as FeatherScanText } from './assets/icons/final/feather-scan-text.js';
export { default as FeatherScan } from './assets/icons/final/feather-scan.js';
export { default as FeatherScatterChart } from './assets/icons/final/feather-scatter-chart.js';
export { default as FeatherSchool2 } from './assets/icons/final/feather-school-2.js';
export { default as FeatherSchool } from './assets/icons/final/feather-school.js';
export { default as FeatherScissorsLineDashed } from './assets/icons/final/feather-scissors-line-dashed.js';
export { default as FeatherScissorsSquareDashedBottom } from './assets/icons/final/feather-scissors-square-dashed-bottom.js';
export { default as FeatherScissorsSquare } from './assets/icons/final/feather-scissors-square.js';
export { default as FeatherScissors } from './assets/icons/final/feather-scissors.js';
export { default as FeatherScreenShareOff } from './assets/icons/final/feather-screen-share-off.js';
export { default as FeatherScreenShare } from './assets/icons/final/feather-screen-share.js';
export { default as FeatherScrollText } from './assets/icons/final/feather-scroll-text.js';
export { default as FeatherScroll } from './assets/icons/final/feather-scroll.js';
export { default as FeatherSearchCheck } from './assets/icons/final/feather-search-check.js';
export { default as FeatherSearchCode } from './assets/icons/final/feather-search-code.js';
export { default as FeatherSearchSlash } from './assets/icons/final/feather-search-slash.js';
export { default as FeatherSearchX } from './assets/icons/final/feather-search-x.js';
export { default as FeatherSearch } from './assets/icons/final/feather-search.js';
export { default as FeatherSection } from './assets/icons/final/feather-section.js';
export { default as FeatherSendHorizontal } from './assets/icons/final/feather-send-horizontal.js';
export { default as FeatherSendToBack } from './assets/icons/final/feather-send-to-back.js';
export { default as FeatherSend } from './assets/icons/final/feather-send.js';
export { default as FeatherSeparatorHorizontal } from './assets/icons/final/feather-separator-horizontal.js';
export { default as FeatherSeparatorVertical } from './assets/icons/final/feather-separator-vertical.js';
export { default as FeatherServerCog } from './assets/icons/final/feather-server-cog.js';
export { default as FeatherServerCrash } from './assets/icons/final/feather-server-crash.js';
export { default as FeatherServerOff } from './assets/icons/final/feather-server-off.js';
export { default as FeatherServer } from './assets/icons/final/feather-server.js';
export { default as FeatherSettings2 } from './assets/icons/final/feather-settings-2.js';
export { default as FeatherSettings } from './assets/icons/final/feather-settings.js';
export { default as FeatherShadowLarge } from './assets/icons/final/feather-shadow-large.js';
export { default as FeatherShadowMedium } from './assets/icons/final/feather-shadow-medium.js';
export { default as FeatherShadowSmall } from './assets/icons/final/feather-shadow-small.js';
export { default as FeatherShapes } from './assets/icons/final/feather-shapes.js';
export { default as FeatherShare2 } from './assets/icons/final/feather-share-2.js';
export { default as FeatherShare } from './assets/icons/final/feather-share.js';
export { default as FeatherSheet } from './assets/icons/final/feather-sheet.js';
export { default as FeatherShell } from './assets/icons/final/feather-shell.js';
export { default as FeatherShieldAlert } from './assets/icons/final/feather-shield-alert.js';
export { default as FeatherShieldBan } from './assets/icons/final/feather-shield-ban.js';
export { default as FeatherShieldCheck } from './assets/icons/final/feather-shield-check.js';
export { default as FeatherShieldClose } from './assets/icons/final/feather-shield-close.js';
export { default as FeatherShieldEllipsis } from './assets/icons/final/feather-shield-ellipsis.js';
export { default as FeatherShieldHalf } from './assets/icons/final/feather-shield-half.js';
export { default as FeatherShieldMinus } from './assets/icons/final/feather-shield-minus.js';
export { default as FeatherShieldOff } from './assets/icons/final/feather-shield-off.js';
export { default as FeatherShieldPlus } from './assets/icons/final/feather-shield-plus.js';
export { default as FeatherShieldQuestionMark } from './assets/icons/final/feather-shield-question-mark.js';
export { default as FeatherShieldQuestion } from './assets/icons/final/feather-shield-question.js';
export { default as FeatherShieldUser } from './assets/icons/final/feather-shield-user.js';
export { default as FeatherShieldX } from './assets/icons/final/feather-shield-x.js';
export { default as FeatherShield } from './assets/icons/final/feather-shield.js';
export { default as FeatherShipWheel } from './assets/icons/final/feather-ship-wheel.js';
export { default as FeatherShip } from './assets/icons/final/feather-ship.js';
export { default as FeatherShirt } from './assets/icons/final/feather-shirt.js';
export { default as FeatherShoppingBag } from './assets/icons/final/feather-shopping-bag.js';
export { default as FeatherShoppingBasket } from './assets/icons/final/feather-shopping-basket.js';
export { default as FeatherShoppingCart } from './assets/icons/final/feather-shopping-cart.js';
export { default as FeatherShovel } from './assets/icons/final/feather-shovel.js';
export { default as FeatherShowerHead } from './assets/icons/final/feather-shower-head.js';
export { default as FeatherShredder } from './assets/icons/final/feather-shredder.js';
export { default as FeatherShrimp } from './assets/icons/final/feather-shrimp.js';
export { default as FeatherShrink } from './assets/icons/final/feather-shrink.js';
export { default as FeatherShrub } from './assets/icons/final/feather-shrub.js';
export { default as FeatherShuffle } from './assets/icons/final/feather-shuffle.js';
export { default as FeatherSidebarClose } from './assets/icons/final/feather-sidebar-close.js';
export { default as FeatherSidebarOpen } from './assets/icons/final/feather-sidebar-open.js';
export { default as FeatherSidebar } from './assets/icons/final/feather-sidebar.js';
export { default as FeatherSigmaSquare } from './assets/icons/final/feather-sigma-square.js';
export { default as FeatherSigma } from './assets/icons/final/feather-sigma.js';
export { default as FeatherSignalHigh } from './assets/icons/final/feather-signal-high.js';
export { default as FeatherSignalLow } from './assets/icons/final/feather-signal-low.js';
export { default as FeatherSignalMedium } from './assets/icons/final/feather-signal-medium.js';
export { default as FeatherSignalZero } from './assets/icons/final/feather-signal-zero.js';
export { default as FeatherSignal } from './assets/icons/final/feather-signal.js';
export { default as FeatherSignature } from './assets/icons/final/feather-signature.js';
export { default as FeatherSignpostBig } from './assets/icons/final/feather-signpost-big.js';
export { default as FeatherSignpost } from './assets/icons/final/feather-signpost.js';
export { default as FeatherSiren } from './assets/icons/final/feather-siren.js';
export { default as FeatherSkipBack } from './assets/icons/final/feather-skip-back.js';
export { default as FeatherSkipForward } from './assets/icons/final/feather-skip-forward.js';
export { default as FeatherSkull } from './assets/icons/final/feather-skull.js';
export { default as FeatherSlack } from './assets/icons/final/feather-slack.js';
export { default as FeatherSlash } from './assets/icons/final/feather-slash.js';
export { default as FeatherSlice } from './assets/icons/final/feather-slice.js';
export { default as FeatherSlidersHorizontal } from './assets/icons/final/feather-sliders-horizontal.js';
export { default as FeatherSliders } from './assets/icons/final/feather-sliders.js';
export { default as FeatherSmartphoneCharging } from './assets/icons/final/feather-smartphone-charging.js';
export { default as FeatherSmartphoneNfc } from './assets/icons/final/feather-smartphone-nfc.js';
export { default as FeatherSmartphone } from './assets/icons/final/feather-smartphone.js';
export { default as FeatherSmilePlus } from './assets/icons/final/feather-smile-plus.js';
export { default as FeatherSmile } from './assets/icons/final/feather-smile.js';
export { default as FeatherSnail } from './assets/icons/final/feather-snail.js';
export { default as FeatherSnowflake } from './assets/icons/final/feather-snowflake.js';
export { default as FeatherSoapDispenserDroplet } from './assets/icons/final/feather-soap-dispenser-droplet.js';
export { default as FeatherSofa } from './assets/icons/final/feather-sofa.js';
export { default as FeatherSortAsc } from './assets/icons/final/feather-sort-asc.js';
export { default as FeatherSortDesc } from './assets/icons/final/feather-sort-desc.js';
export { default as FeatherSoup } from './assets/icons/final/feather-soup.js';
export { default as FeatherSpace } from './assets/icons/final/feather-space.js';
export { default as FeatherSpade } from './assets/icons/final/feather-spade.js';
export { default as FeatherSparkle } from './assets/icons/final/feather-sparkle.js';
export { default as FeatherSparkles } from './assets/icons/final/feather-sparkles.js';
export { default as FeatherSpeaker } from './assets/icons/final/feather-speaker.js';
export { default as FeatherSpeech } from './assets/icons/final/feather-speech.js';
export { default as FeatherSpellCheck2 } from './assets/icons/final/feather-spell-check-2.js';
export { default as FeatherSpellCheck } from './assets/icons/final/feather-spell-check.js';
export { default as FeatherSplinePointer } from './assets/icons/final/feather-spline-pointer.js';
export { default as FeatherSpline } from './assets/icons/final/feather-spline.js';
export { default as FeatherSplitSquareHorizontal } from './assets/icons/final/feather-split-square-horizontal.js';
export { default as FeatherSplitSquareVertical } from './assets/icons/final/feather-split-square-vertical.js';
export { default as FeatherSplit } from './assets/icons/final/feather-split.js';
export { default as FeatherSpool } from './assets/icons/final/feather-spool.js';
export { default as FeatherSprayCan } from './assets/icons/final/feather-spray-can.js';
export { default as FeatherSprout } from './assets/icons/final/feather-sprout.js';
export { default as FeatherSquareArrowOutDownLeft } from './assets/icons/final/feather-square-arrow-out-down-left.js';
export { default as FeatherSquareArrowOutDownRight } from './assets/icons/final/feather-square-arrow-out-down-right.js';
export { default as FeatherSquareArrowOutUpLeft } from './assets/icons/final/feather-square-arrow-out-up-left.js';
export { default as FeatherSquareArrowOutUpRight } from './assets/icons/final/feather-square-arrow-out-up-right.js';
export { default as FeatherSquareAsterisk } from './assets/icons/final/feather-square-asterisk.js';
export { default as FeatherSquareCheckBig } from './assets/icons/final/feather-square-check-big.js';
export { default as FeatherSquareCheck } from './assets/icons/final/feather-square-check.js';
export { default as FeatherSquareCode } from './assets/icons/final/feather-square-code.js';
export { default as FeatherSquareCornerBl } from './assets/icons/final/feather-square-corner-bl.js';
export { default as FeatherSquareCornerBrBl } from './assets/icons/final/feather-square-corner-br-bl.js';
export { default as FeatherSquareCornerBr } from './assets/icons/final/feather-square-corner-br.js';
export { default as FeatherSquareCornerTlBl } from './assets/icons/final/feather-square-corner-tl-bl.js';
export { default as FeatherSquareCornerTlBrBl } from './assets/icons/final/feather-square-corner-tl-br-bl.js';
export { default as FeatherSquareCornerTlBr } from './assets/icons/final/feather-square-corner-tl-br.js';
export { default as FeatherSquareCornerTlTrBl } from './assets/icons/final/feather-square-corner-tl-tr-bl.js';
export { default as FeatherSquareCornerTlTrBr } from './assets/icons/final/feather-square-corner-tl-tr-br.js';
export { default as FeatherSquareCornerTl } from './assets/icons/final/feather-square-corner-tl.js';
export { default as FeatherSquareCornerTrBl } from './assets/icons/final/feather-square-corner-tr-bl.js';
export { default as FeatherSquareCornerTrBrBl } from './assets/icons/final/feather-square-corner-tr-br-bl.js';
export { default as FeatherSquareCornerTrBr } from './assets/icons/final/feather-square-corner-tr-br.js';
export { default as FeatherSquareCornerTrTl } from './assets/icons/final/feather-square-corner-tr-tl.js';
export { default as FeatherSquareCornerTr } from './assets/icons/final/feather-square-corner-tr.js';
export { default as FeatherSquareCorner } from './assets/icons/final/feather-square-corner.js';
export { default as FeatherSquareDashedBottomCode } from './assets/icons/final/feather-square-dashed-bottom-code.js';
export { default as FeatherSquareDashedBottom } from './assets/icons/final/feather-square-dashed-bottom.js';
export { default as FeatherSquareDashedMousePointer } from './assets/icons/final/feather-square-dashed-mouse-pointer.js';
export { default as FeatherSquareDashedTopSolid } from './assets/icons/final/feather-square-dashed-top-solid.js';
export { default as FeatherSquareDot } from './assets/icons/final/feather-square-dot.js';
export { default as FeatherSquareEqual } from './assets/icons/final/feather-square-equal.js';
export { default as FeatherSquareMousePointer } from './assets/icons/final/feather-square-mouse-pointer.js';
export { default as FeatherSquarePen } from './assets/icons/final/feather-square-pen.js';
export { default as FeatherSquarePower } from './assets/icons/final/feather-square-power.js';
export { default as FeatherSquareRadical } from './assets/icons/final/feather-square-radical.js';
export { default as FeatherSquareRoundCorner } from './assets/icons/final/feather-square-round-corner.js';
export { default as FeatherSquareSidesBottom } from './assets/icons/final/feather-square-sides-bottom.js';
export { default as FeatherSquareSidesLeftBottom } from './assets/icons/final/feather-square-sides-left-bottom.js';
export { default as FeatherSquareSidesLeftRightBottom } from './assets/icons/final/feather-square-sides-left-right-bottom.js';
export { default as FeatherSquareSidesLeftRight } from './assets/icons/final/feather-square-sides-left-right.js';
export { default as FeatherSquareSidesLeftTopBottom } from './assets/icons/final/feather-square-sides-left-top-bottom.js';
export { default as FeatherSquareSidesLeftTopRight } from './assets/icons/final/feather-square-sides-left-top-right.js';
export { default as FeatherSquareSidesLeftTop } from './assets/icons/final/feather-square-sides-left-top.js';
export { default as FeatherSquareSidesLeft } from './assets/icons/final/feather-square-sides-left.js';
export { default as FeatherSquareSidesRightBottom } from './assets/icons/final/feather-square-sides-right-bottom.js';
export { default as FeatherSquareSidesRight } from './assets/icons/final/feather-square-sides-right.js';
export { default as FeatherSquareSidesTopBottom } from './assets/icons/final/feather-square-sides-top-bottom.js';
export { default as FeatherSquareSidesTopRightBottom } from './assets/icons/final/feather-square-sides-top-right-bottom.js';
export { default as FeatherSquareSidesTopRight } from './assets/icons/final/feather-square-sides-top-right.js';
export { default as FeatherSquareSidesTop } from './assets/icons/final/feather-square-sides-top.js';
export { default as FeatherSquareSides } from './assets/icons/final/feather-square-sides.js';
export { default as FeatherSquareSlash } from './assets/icons/final/feather-square-slash.js';
export { default as FeatherSquareSquare } from './assets/icons/final/feather-square-square.js';
export { default as FeatherSquareStack } from './assets/icons/final/feather-square-stack.js';
export { default as FeatherSquare } from './assets/icons/final/feather-square.js';
export { default as FeatherSquaresExclude } from './assets/icons/final/feather-squares-exclude.js';
export { default as FeatherSquaresIntersect } from './assets/icons/final/feather-squares-intersect.js';
export { default as FeatherSquaresSubtract } from './assets/icons/final/feather-squares-subtract.js';
export { default as FeatherSquaresUnite } from './assets/icons/final/feather-squares-unite.js';
export { default as FeatherSquircleDashed } from './assets/icons/final/feather-squircle-dashed.js';
export { default as FeatherSquircle } from './assets/icons/final/feather-squircle.js';
export { default as FeatherSquirrel } from './assets/icons/final/feather-squirrel.js';
export { default as FeatherStamp } from './assets/icons/final/feather-stamp.js';
export { default as FeatherStarHalf } from './assets/icons/final/feather-star-half.js';
export { default as FeatherStarOff } from './assets/icons/final/feather-star-off.js';
export { default as FeatherStar } from './assets/icons/final/feather-star.js';
export { default as FeatherStepBack } from './assets/icons/final/feather-step-back.js';
export { default as FeatherStepForward } from './assets/icons/final/feather-step-forward.js';
export { default as FeatherStethoscope } from './assets/icons/final/feather-stethoscope.js';
export { default as FeatherSticker } from './assets/icons/final/feather-sticker.js';
export { default as FeatherStickyNote } from './assets/icons/final/feather-sticky-note.js';
export { default as FeatherStopCircle } from './assets/icons/final/feather-stop-circle.js';
export { default as FeatherStore } from './assets/icons/final/feather-store.js';
export { default as FeatherStretchHorizontal } from './assets/icons/final/feather-stretch-horizontal.js';
export { default as FeatherStretchVertical } from './assets/icons/final/feather-stretch-vertical.js';
export { default as FeatherStrikethrough } from './assets/icons/final/feather-strikethrough.js';
export { default as FeatherSubframe } from './assets/icons/final/feather-subframe.js';
export { default as FeatherSubscript } from './assets/icons/final/feather-subscript.js';
export { default as FeatherSubtitles } from './assets/icons/final/feather-subtitles.js';
export { default as FeatherSunDim } from './assets/icons/final/feather-sun-dim.js';
export { default as FeatherSunMedium } from './assets/icons/final/feather-sun-medium.js';
export { default as FeatherSunMoon } from './assets/icons/final/feather-sun-moon.js';
export { default as FeatherSunSnow } from './assets/icons/final/feather-sun-snow.js';
export { default as FeatherSun } from './assets/icons/final/feather-sun.js';
export { default as FeatherSunrise } from './assets/icons/final/feather-sunrise.js';
export { default as FeatherSunset } from './assets/icons/final/feather-sunset.js';
export { default as FeatherSuperscript } from './assets/icons/final/feather-superscript.js';
export { default as FeatherSwatchBook } from './assets/icons/final/feather-swatch-book.js';
export { default as FeatherSwissFranc } from './assets/icons/final/feather-swiss-franc.js';
export { default as FeatherSwitchCamera } from './assets/icons/final/feather-switch-camera.js';
export { default as FeatherSword } from './assets/icons/final/feather-sword.js';
export { default as FeatherSwords } from './assets/icons/final/feather-swords.js';
export { default as FeatherSyringe } from './assets/icons/final/feather-syringe.js';
export { default as FeatherTable2 } from './assets/icons/final/feather-table-2.js';
export { default as FeatherTableCellsMerge } from './assets/icons/final/feather-table-cells-merge.js';
export { default as FeatherTableCellsSplit } from './assets/icons/final/feather-table-cells-split.js';
export { default as FeatherTableColumnsSplit } from './assets/icons/final/feather-table-columns-split.js';
export { default as FeatherTableOfContents } from './assets/icons/final/feather-table-of-contents.js';
export { default as FeatherTableProperties } from './assets/icons/final/feather-table-properties.js';
export { default as FeatherTableRowsSplit } from './assets/icons/final/feather-table-rows-split.js';
export { default as FeatherTable } from './assets/icons/final/feather-table.js';
export { default as FeatherTabletSmartphone } from './assets/icons/final/feather-tablet-smartphone.js';
export { default as FeatherTablet } from './assets/icons/final/feather-tablet.js';
export { default as FeatherTablets } from './assets/icons/final/feather-tablets.js';
export { default as FeatherTag } from './assets/icons/final/feather-tag.js';
export { default as FeatherTags } from './assets/icons/final/feather-tags.js';
export { default as FeatherTally1 } from './assets/icons/final/feather-tally-1.js';
export { default as FeatherTally2 } from './assets/icons/final/feather-tally-2.js';
export { default as FeatherTally3 } from './assets/icons/final/feather-tally-3.js';
export { default as FeatherTally4 } from './assets/icons/final/feather-tally-4.js';
export { default as FeatherTally5 } from './assets/icons/final/feather-tally-5.js';
export { default as FeatherTangent } from './assets/icons/final/feather-tangent.js';
export { default as FeatherTarget } from './assets/icons/final/feather-target.js';
export { default as FeatherTelescope } from './assets/icons/final/feather-telescope.js';
export { default as FeatherTentTree } from './assets/icons/final/feather-tent-tree.js';
export { default as FeatherTent } from './assets/icons/final/feather-tent.js';
export { default as FeatherTerminalSquare } from './assets/icons/final/feather-terminal-square.js';
export { default as FeatherTerminal } from './assets/icons/final/feather-terminal.js';
export { default as FeatherTestTube2 } from './assets/icons/final/feather-test-tube-2.js';
export { default as FeatherTestTubeDiagonal } from './assets/icons/final/feather-test-tube-diagonal.js';
export { default as FeatherTestTube } from './assets/icons/final/feather-test-tube.js';
export { default as FeatherTestTubes } from './assets/icons/final/feather-test-tubes.js';
export { default as FeatherTextCursorInput } from './assets/icons/final/feather-text-cursor-input.js';
export { default as FeatherTextCursor } from './assets/icons/final/feather-text-cursor.js';
export { default as FeatherTextQuote } from './assets/icons/final/feather-text-quote.js';
export { default as FeatherTextSearch } from './assets/icons/final/feather-text-search.js';
export { default as FeatherTextSelect } from './assets/icons/final/feather-text-select.js';
export { default as FeatherTextSize } from './assets/icons/final/feather-text-size.js';
export { default as FeatherTextTruncate } from './assets/icons/final/feather-text-truncate.js';
export { default as FeatherText } from './assets/icons/final/feather-text.js';
export { default as FeatherTheater } from './assets/icons/final/feather-theater.js';
export { default as FeatherThermometerSnowflake } from './assets/icons/final/feather-thermometer-snowflake.js';
export { default as FeatherThermometerSun } from './assets/icons/final/feather-thermometer-sun.js';
export { default as FeatherThermometer } from './assets/icons/final/feather-thermometer.js';
export { default as FeatherThumbsDown } from './assets/icons/final/feather-thumbs-down.js';
export { default as FeatherThumbsUp } from './assets/icons/final/feather-thumbs-up.js';
export { default as FeatherTicketCheck } from './assets/icons/final/feather-ticket-check.js';
export { default as FeatherTicketMinus } from './assets/icons/final/feather-ticket-minus.js';
export { default as FeatherTicketPercent } from './assets/icons/final/feather-ticket-percent.js';
export { default as FeatherTicketPlus } from './assets/icons/final/feather-ticket-plus.js';
export { default as FeatherTicketSlash } from './assets/icons/final/feather-ticket-slash.js';
export { default as FeatherTicketX } from './assets/icons/final/feather-ticket-x.js';
export { default as FeatherTicket } from './assets/icons/final/feather-ticket.js';
export { default as FeatherTicketsPlane } from './assets/icons/final/feather-tickets-plane.js';
export { default as FeatherTickets } from './assets/icons/final/feather-tickets.js';
export { default as FeatherTimerOff } from './assets/icons/final/feather-timer-off.js';
export { default as FeatherTimerReset } from './assets/icons/final/feather-timer-reset.js';
export { default as FeatherTimer } from './assets/icons/final/feather-timer.js';
export { default as FeatherToggleLeft } from './assets/icons/final/feather-toggle-left.js';
export { default as FeatherToggleRight } from './assets/icons/final/feather-toggle-right.js';
export { default as FeatherToilet } from './assets/icons/final/feather-toilet.js';
export { default as FeatherTool } from './assets/icons/final/feather-tool.js';
export { default as FeatherTornado } from './assets/icons/final/feather-tornado.js';
export { default as FeatherTorus } from './assets/icons/final/feather-torus.js';
export { default as FeatherTouchpadOff } from './assets/icons/final/feather-touchpad-off.js';
export { default as FeatherTouchpad } from './assets/icons/final/feather-touchpad.js';
export { default as FeatherTowerControl } from './assets/icons/final/feather-tower-control.js';
export { default as FeatherToyBrick } from './assets/icons/final/feather-toy-brick.js';
export { default as FeatherTractor } from './assets/icons/final/feather-tractor.js';
export { default as FeatherTrafficCone } from './assets/icons/final/feather-traffic-cone.js';
export { default as FeatherTrainFrontTunnel } from './assets/icons/final/feather-train-front-tunnel.js';
export { default as FeatherTrainFront } from './assets/icons/final/feather-train-front.js';
export { default as FeatherTrainTrack } from './assets/icons/final/feather-train-track.js';
export { default as FeatherTrain } from './assets/icons/final/feather-train.js';
export { default as FeatherTramFront } from './assets/icons/final/feather-tram-front.js';
export { default as FeatherTransgender } from './assets/icons/final/feather-transgender.js';
export { default as FeatherTrash2 } from './assets/icons/final/feather-trash-2.js';
export { default as FeatherTrash } from './assets/icons/final/feather-trash.js';
export { default as FeatherTreeDeciduous } from './assets/icons/final/feather-tree-deciduous.js';
export { default as FeatherTreePalm } from './assets/icons/final/feather-tree-palm.js';
export { default as FeatherTreePine } from './assets/icons/final/feather-tree-pine.js';
export { default as FeatherTrees } from './assets/icons/final/feather-trees.js';
export { default as FeatherTrello } from './assets/icons/final/feather-trello.js';
export { default as FeatherTrendingDown } from './assets/icons/final/feather-trending-down.js';
export { default as FeatherTrendingUpDown } from './assets/icons/final/feather-trending-up-down.js';
export { default as FeatherTrendingUp } from './assets/icons/final/feather-trending-up.js';
export { default as FeatherTriangleAlert } from './assets/icons/final/feather-triangle-alert.js';
export { default as FeatherTriangleDashed } from './assets/icons/final/feather-triangle-dashed.js';
export { default as FeatherTriangleRight } from './assets/icons/final/feather-triangle-right.js';
export { default as FeatherTriangle } from './assets/icons/final/feather-triangle.js';
export { default as FeatherTrophy } from './assets/icons/final/feather-trophy.js';
export { default as FeatherTruckElectric } from './assets/icons/final/feather-truck-electric.js';
export { default as FeatherTruck } from './assets/icons/final/feather-truck.js';
export { default as FeatherTurtle } from './assets/icons/final/feather-turtle.js';
export { default as FeatherTv2 } from './assets/icons/final/feather-tv-2.js';
export { default as FeatherTvMinimalPlay } from './assets/icons/final/feather-tv-minimal-play.js';
export { default as FeatherTv } from './assets/icons/final/feather-tv.js';
export { default as FeatherTwitch } from './assets/icons/final/feather-twitch.js';
export { default as FeatherTwitter } from './assets/icons/final/feather-twitter.js';
export { default as FeatherTypeOutline } from './assets/icons/final/feather-type-outline.js';
export { default as FeatherType } from './assets/icons/final/feather-type.js';
export { default as FeatherUmbrellaOff } from './assets/icons/final/feather-umbrella-off.js';
export { default as FeatherUmbrella } from './assets/icons/final/feather-umbrella.js';
export { default as FeatherUnderline } from './assets/icons/final/feather-underline.js';
export { default as FeatherUndo2 } from './assets/icons/final/feather-undo-2.js';
export { default as FeatherUndoDot } from './assets/icons/final/feather-undo-dot.js';
export { default as FeatherUndo } from './assets/icons/final/feather-undo.js';
export { default as FeatherUnfoldHorizontal } from './assets/icons/final/feather-unfold-horizontal.js';
export { default as FeatherUnfoldVertical } from './assets/icons/final/feather-unfold-vertical.js';
export { default as FeatherUngroup } from './assets/icons/final/feather-ungroup.js';
export { default as FeatherUniversity } from './assets/icons/final/feather-university.js';
export { default as FeatherUnlink2 } from './assets/icons/final/feather-unlink-2.js';
export { default as FeatherUnlink } from './assets/icons/final/feather-unlink.js';
export { default as FeatherUnlockKeyhole } from './assets/icons/final/feather-unlock-keyhole.js';
export { default as FeatherUnlock } from './assets/icons/final/feather-unlock.js';
export { default as FeatherUnplug } from './assets/icons/final/feather-unplug.js';
export { default as FeatherUploadCloud } from './assets/icons/final/feather-upload-cloud.js';
export { default as FeatherUpload } from './assets/icons/final/feather-upload.js';
export { default as FeatherUsb } from './assets/icons/final/feather-usb.js';
export { default as FeatherUser2 } from './assets/icons/final/feather-user-2.js';
export { default as FeatherUserCheck2 } from './assets/icons/final/feather-user-check-2.js';
export { default as FeatherUserCheck } from './assets/icons/final/feather-user-check.js';
export { default as FeatherUserCircle2 } from './assets/icons/final/feather-user-circle-2.js';
export { default as FeatherUserCircle } from './assets/icons/final/feather-user-circle.js';
export { default as FeatherUserCog2 } from './assets/icons/final/feather-user-cog-2.js';
export { default as FeatherUserCog } from './assets/icons/final/feather-user-cog.js';
export { default as FeatherUserLock } from './assets/icons/final/feather-user-lock.js';
export { default as FeatherUserMinus2 } from './assets/icons/final/feather-user-minus-2.js';
export { default as FeatherUserMinus } from './assets/icons/final/feather-user-minus.js';
export { default as FeatherUserPen } from './assets/icons/final/feather-user-pen.js';
export { default as FeatherUserPlus2 } from './assets/icons/final/feather-user-plus-2.js';
export { default as FeatherUserPlus } from './assets/icons/final/feather-user-plus.js';
export { default as FeatherUserRoundCheck } from './assets/icons/final/feather-user-round-check.js';
export { default as FeatherUserRoundCog } from './assets/icons/final/feather-user-round-cog.js';
export { default as FeatherUserRoundMinus } from './assets/icons/final/feather-user-round-minus.js';
export { default as FeatherUserRoundPen } from './assets/icons/final/feather-user-round-pen.js';
export { default as FeatherUserRoundPlus } from './assets/icons/final/feather-user-round-plus.js';
export { default as FeatherUserRoundSearch } from './assets/icons/final/feather-user-round-search.js';
export { default as FeatherUserRoundX } from './assets/icons/final/feather-user-round-x.js';
export { default as FeatherUserRound } from './assets/icons/final/feather-user-round.js';
export { default as FeatherUserSearch } from './assets/icons/final/feather-user-search.js';
export { default as FeatherUserSquare2 } from './assets/icons/final/feather-user-square-2.js';
export { default as FeatherUserSquare } from './assets/icons/final/feather-user-square.js';
export { default as FeatherUserX2 } from './assets/icons/final/feather-user-x-2.js';
export { default as FeatherUserX } from './assets/icons/final/feather-user-x.js';
export { default as FeatherUser } from './assets/icons/final/feather-user.js';
export { default as FeatherUsers2 } from './assets/icons/final/feather-users-2.js';
export { default as FeatherUsersRound } from './assets/icons/final/feather-users-round.js';
export { default as FeatherUsers } from './assets/icons/final/feather-users.js';
export { default as FeatherUtensilsCrossed } from './assets/icons/final/feather-utensils-crossed.js';
export { default as FeatherUtensils } from './assets/icons/final/feather-utensils.js';
export { default as FeatherUtilityPole } from './assets/icons/final/feather-utility-pole.js';
export { default as FeatherVariable } from './assets/icons/final/feather-variable.js';
export { default as FeatherVault } from './assets/icons/final/feather-vault.js';
export { default as FeatherVegan } from './assets/icons/final/feather-vegan.js';
export { default as FeatherVenetianMask } from './assets/icons/final/feather-venetian-mask.js';
export { default as FeatherVenusAndMars } from './assets/icons/final/feather-venus-and-mars.js';
export { default as FeatherVenus } from './assets/icons/final/feather-venus.js';
export { default as FeatherVerified } from './assets/icons/final/feather-verified.js';
export { default as FeatherVibrateOff } from './assets/icons/final/feather-vibrate-off.js';
export { default as FeatherVibrate } from './assets/icons/final/feather-vibrate.js';
export { default as FeatherVideoOff } from './assets/icons/final/feather-video-off.js';
export { default as FeatherVideo } from './assets/icons/final/feather-video.js';
export { default as FeatherVideotape } from './assets/icons/final/feather-videotape.js';
export { default as FeatherView } from './assets/icons/final/feather-view.js';
export { default as FeatherVoicemail } from './assets/icons/final/feather-voicemail.js';
export { default as FeatherVolleyball } from './assets/icons/final/feather-volleyball.js';
export { default as FeatherVolume1 } from './assets/icons/final/feather-volume-1.js';
export { default as FeatherVolume2 } from './assets/icons/final/feather-volume-2.js';
export { default as FeatherVolumeOff } from './assets/icons/final/feather-volume-off.js';
export { default as FeatherVolumeX } from './assets/icons/final/feather-volume-x.js';
export { default as FeatherVolume } from './assets/icons/final/feather-volume.js';
export { default as FeatherVote } from './assets/icons/final/feather-vote.js';
export { default as FeatherWallet2 } from './assets/icons/final/feather-wallet-2.js';
export { default as FeatherWalletCards } from './assets/icons/final/feather-wallet-cards.js';
export { default as FeatherWallet } from './assets/icons/final/feather-wallet.js';
export { default as FeatherWallpaper } from './assets/icons/final/feather-wallpaper.js';
export { default as FeatherWand2 } from './assets/icons/final/feather-wand-2.js';
export { default as FeatherWandSparkles } from './assets/icons/final/feather-wand-sparkles.js';
export { default as FeatherWand } from './assets/icons/final/feather-wand.js';
export { default as FeatherWarehouse } from './assets/icons/final/feather-warehouse.js';
export { default as FeatherWashingMachine } from './assets/icons/final/feather-washing-machine.js';
export { default as FeatherWatch } from './assets/icons/final/feather-watch.js';
export { default as FeatherWavesLadder } from './assets/icons/final/feather-waves-ladder.js';
export { default as FeatherWaves } from './assets/icons/final/feather-waves.js';
export { default as FeatherWaypoints } from './assets/icons/final/feather-waypoints.js';
export { default as FeatherWebcam } from './assets/icons/final/feather-webcam.js';
export { default as FeatherWebhookOff } from './assets/icons/final/feather-webhook-off.js';
export { default as FeatherWebhook } from './assets/icons/final/feather-webhook.js';
export { default as FeatherWeight } from './assets/icons/final/feather-weight.js';
export { default as FeatherWheatOff } from './assets/icons/final/feather-wheat-off.js';
export { default as FeatherWheat } from './assets/icons/final/feather-wheat.js';
export { default as FeatherWholeWord } from './assets/icons/final/feather-whole-word.js';
export { default as FeatherWifiCog } from './assets/icons/final/feather-wifi-cog.js';
export { default as FeatherWifiHigh } from './assets/icons/final/feather-wifi-high.js';
export { default as FeatherWifiLow } from './assets/icons/final/feather-wifi-low.js';
export { default as FeatherWifiOff } from './assets/icons/final/feather-wifi-off.js';
export { default as FeatherWifiPen } from './assets/icons/final/feather-wifi-pen.js';
export { default as FeatherWifiZero } from './assets/icons/final/feather-wifi-zero.js';
export { default as FeatherWifi } from './assets/icons/final/feather-wifi.js';
export { default as FeatherWindArrowDown } from './assets/icons/final/feather-wind-arrow-down.js';
export { default as FeatherWind } from './assets/icons/final/feather-wind.js';
export { default as FeatherWineOff } from './assets/icons/final/feather-wine-off.js';
export { default as FeatherWine } from './assets/icons/final/feather-wine.js';
export { default as FeatherWorkflow } from './assets/icons/final/feather-workflow.js';
export { default as FeatherWorm } from './assets/icons/final/feather-worm.js';
export { default as FeatherWrapText } from './assets/icons/final/feather-wrap-text.js';
export { default as FeatherWrench } from './assets/icons/final/feather-wrench.js';
export { default as FeatherXCircle } from './assets/icons/final/feather-x-circle.js';
export { default as FeatherXOctagon } from './assets/icons/final/feather-x-octagon.js';
export { default as FeatherXSquare } from './assets/icons/final/feather-x-square.js';
export { default as FeatherXTwitter } from './assets/icons/final/feather-x-twitter.js';
export { default as FeatherX } from './assets/icons/final/feather-x.js';
export { default as FeatherYoutube } from './assets/icons/final/feather-youtube.js';
export { default as FeatherZapOff } from './assets/icons/final/feather-zap-off.js';
export { default as FeatherZap } from './assets/icons/final/feather-zap.js';
export { default as FeatherZoomIn } from './assets/icons/final/feather-zoom-in.js';
export { default as FeatherZoomOut } from './assets/icons/final/feather-zoom-out.js';
export { default as SubframeSides } from './assets/icons/final/subframe-sides.js';

/**
 * Finally, create our default export using the imported values
 */
var SubframeCore = __assign(__assign({}, index), { RadioGroup: radioGroup, Checkbox: checkbox, Popover: popover, HoverCard: hoverCard, Tooltip: tooltip, DropdownMenu: dropdownMenu, ContextMenu: contextMenu, Toaster: Oe, toast: ce, Progress: progress, Slider: slider, Select: select, AreaChart: AreaChart, BarChart: BarChart, LineChart: LineChart, PieChart: PieChart, CartesianGrid: CartesianGrid, XAxis: XAxis, YAxis: YAxis, ChartTooltip: ChartTooltip, ChartLegend: ChartLegend, ChartLabel: ChartLabel, Calendar: Calendar, Skeleton: Skeleton, CopyToClipboard: copyToClipboard, Switch: _switch, Loader: Loader, Icon: Icon, IconWrapper: IconWrapper, Dialog: dialog, Drawer: drawer, FullScreenDialog: fullscreenDialog, ToggleGroup: toggleGroup, Collapsible: collapsible, Text: Text, TypescriptHelpers: typescriptHelpers, createTwClassNames: createTwClassNames, twClassNames: twClassNames });

export { AreaChart, BarChart, Calendar, CartesianGrid, ChartLabel, ChartLegend, ChartTooltip, checkbox as Checkbox, collapsible as Collapsible, contextMenu as ContextMenu, copyToClipboard as CopyToClipboard, dialog as Dialog, drawer as Drawer, dropdownMenu as DropdownMenu, fullscreenDialog as FullScreenDialog, hoverCard as HoverCard, Icon, IconWrapper, LineChart, Loader, PieChart, popover as Popover, progress as Progress, radioGroup as RadioGroup, select as Select, Skeleton, slider as Slider, _switch as Switch, Text, Oe as Toaster, toggleGroup as ToggleGroup, tooltip as Tooltip, typescriptHelpers as TypescriptHelpers, XAxis, YAxis, createTwClassNames, SubframeCore as default, ce as toast, twClassNames };
