// @subframe/core/dist/cjs/external/d3-time/src/week.js
'use strict';

var interval = require('./interval.js');
var duration = require('./duration.js');

function timeWeekday(i) {
  return interval.timeInterval((date) => {
    date.setDate(date.getDate() - (date.getDay() + 7 - i) % 7);
    date.setHours(0, 0, 0, 0);
  }, (date, step) => {
    date.setDate(date.getDate() + step * 7);
  }, (start, end) => {
    return (end - start - (end.getTimezoneOffset() - start.getTimezoneOffset()) * duration.durationMinute) / duration.durationWeek;
  });
}

const timeSunday = timeWeekday(0);
const timeMonday = timeWeekday(1);
const timeTuesday = timeWeekday(2);
const timeWednesday = timeWeekday(3);
const timeThursday = timeWeekday(4);
const timeFriday = timeWeekday(5);
const timeSaturday = timeWeekday(6);

timeSunday.range;
timeMonday.range;
timeTuesday.range;
timeWednesday.range;
timeThursday.range;
timeFriday.range;
timeSaturday.range;

function utcWeekday(i) {
  return interval.timeInterval((date) => {
    date.setUTCDate(date.getUTCDate() - (date.getUTCDay() + 7 - i) % 7);
    date.setUTCHours(0, 0, 0, 0);
  }, (date, step) => {
    date.setUTCDate(date.getUTCDate() + step * 7);
  }, (start, end) => {
    return (end - start) / duration.durationWeek;
  });
}

const utcSunday = utcWeekday(0);
const utcMonday = utcWeekday(1);
const utcTuesday = utcWeekday(2);
const utcWednesday = utcWeekday(3);
const utcThursday = utcWeekday(4);
const utcFriday = utcWeekday(5);
const utcSaturday = utcWeekday(6);

utcSunday.range;
utcMonday.range;
utcTuesday.range;
utcWednesday.range;
utcThursday.range;
utcFriday.range;
utcSaturday.range;

exports.timeFriday = timeFriday;
exports.timeMonday = timeMonday;
exports.timeSaturday = timeSaturday;
exports.timeSunday = timeSunday;
exports.timeThursday = timeThursday;
exports.timeTuesday = timeTuesday;
exports.timeWednesday = timeWednesday;
exports.utcFriday = utcFriday;
exports.utcMonday = utcMonday;
exports.utcSaturday = utcSaturday;
exports.utcSunday = utcSunday;
exports.utcThursday = utcThursday;
exports.utcTuesday = utcTuesday;
exports.utcWednesday = utcWednesday;
